<!DOCTYPE html>
<html>

  <head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  <title>Molecular QTL study</title>
  <meta name="description" content="A collection of pulication/dataset/cohort about molecular QTL study.">

  <link rel="stylesheet" href="/css/main.css">
  <link rel="canonical" href="http://localhost:4000/2018/10/03/molecular-qtl.html">
  <link rel="alternate" type="application/rss+xml" title="Huanwei Wang's blog" href="http://localhost:4000/feed.xml">
</head>


  <body>

    <!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-121065275-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-121065275-1');
</script>



    <header class="site-header">

  <div class="wrapper">

    <a class="site-title" href="/">Huanwei Wang's blog</a>

    <nav class="site-nav">
      <a href="#" class="menu-icon">
        <svg viewBox="0 0 18 15">
          <path fill="#424242" d="M18,1.484c0,0.82-0.665,1.484-1.484,1.484H1.484C0.665,2.969,0,2.304,0,1.484l0,0C0,0.665,0.665,0,1.484,0 h15.031C17.335,0,18,0.665,18,1.484L18,1.484z"/>
          <path fill="#424242" d="M18,7.516C18,8.335,17.335,9,16.516,9H1.484C0.665,9,0,8.335,0,7.516l0,0c0-0.82,0.665-1.484,1.484-1.484 h15.031C17.335,6.031,18,6.696,18,7.516L18,7.516z"/>
          <path fill="#424242" d="M18,13.516C18,14.335,17.335,15,16.516,15H1.484C0.665,15,0,14.335,0,13.516l0,0 c0-0.82,0.665-1.484,1.484-1.484h15.031C17.335,12.031,18,12.696,18,13.516L18,13.516z"/>
        </svg>
      </a>

      <div class="trigger">
        
          
        
          
          <a class="page-link" href="/about/">About</a>
          
        
          
        
          
        
          
        
          
        
          
        
          
          <a class="page-link" href="/links/">Links</a>
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
          <a class="page-link" href="/tags/">Tags</a>
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
      </div>
    </nav>

  </div>

</header>


    <div class="page-content">
      <div class="wrapper">
        <article class="post" itemscope itemtype="http://schema.org/BlogPosting">

  <header class="post-header">
    <h1 class="post-title" itemprop="name headline">Molecular QTL study</h1>
    <p class="post-meta"><time datetime="2018-10-03T00:00:00+10:00" itemprop="datePublished">Oct 3, 2018</time></p>
  </header>

  <div class="post-content" itemprop="articleBody">
    <p>A collection of pulication/dataset/cohort about molecular QTL study.</p>

<h3 id="atac-qtl-chromatin-accessbility">ATAC-QTL (chromatin accessbility)</h3>

<ul>
  <li>Gate, R. E., Cheng, C. S., Aiden, A. P., Siba, A., Tabaka, M., Lituiev, D., … Regev, A. (2018). Genetic determinants of co-accessible chromatin regions in activated T cells across humans. Nature Genetics, 50(8), 1140–1150. <a href="https://doi.org/10.1038/s41588-018-0156-2">https://doi.org/10.1038/s41588-018-0156-2</a></li>
</ul>

<h3 id="hqtl-histone-marker">hQTL (histone marker)</h3>

<ul>
  <li>McVicker, G., van de Geijn, B., Degner, J. F., Cain, C. E., Banovich, N. E., Raj, A., … Pritchard, J. K. (2013). Identification of Genetic Variants That Affect Histone Modifications in Human Cells. Science, 342(6159), 747–749. <a href="https://doi.org/10.1126/science.1242429">https://doi.org/10.1126/science.1242429</a></li>
  <li>Grubert, F., Zaugg, J. B., Kasowski, M., Ursu, O., Spacek, D. V., Martin, A. R., … Snyder, M. (2015). Genetic Control of Chromatin States in Humans Involves Local and Distal Chromosomal Interactions. Cell, 162(5), 1051–1065. <a href="https://doi.org/10.1016/j.cell.2015.07.048">https://doi.org/10.1016/j.cell.2015.07.048</a></li>
</ul>

<h3 id="mqtl-dna-methylation">mQTL (DNA methylation)</h3>

<ul>
  <li>Meta study of <a href="https://www.lothianbirthcohort.ed.ac.uk/">LBC</a> and <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0035430">BSGS</a>
    <ul>
      <li>McRae, A., Marioni, R. E., Shah, S., Yang, J., Powell, J. E., Harris, S. E., … Montgomery, G. W. (2017). Identification of 55,000 Replicated DNA Methylation QTL. BioRxiv. Retrieved from <a href="http://biorxiv.org/content/early/2017/07/21/166710.abstract">http://biorxiv.org/content/early/2017/07/21/166710.abstract</a></li>
    </ul>
  </li>
</ul>

<h3 id="eqtl-gene-expression">eQTL (gene expression)</h3>

<ul>
  <li><a href="http://cnsgenomics.com/shiny/CAGE/">CAGE study</a>
    <ul>
      <li>Lloyd-Jones, L. R., Holloway, A., McRae, A., Yang, J., Small, K., Zeng, B., … Powell, J. E. (2017). The Genetic Architecture of Gene Expression in Peripheral Blood. The American Journal of Human Genetics, 100(2), 228–237. <a href="https://doi.org/10.1016/j.ajhg.2016.12.008">https://doi.org/10.1016/j.ajhg.2016.12.008</a></li>
    </ul>
  </li>
  <li><a href="https://gtexportal.org/home/">GTEx study</a>
    <ul>
      <li>Aguet, F., Ardlie, K. G., Cummings, B. B., Gelfand, E. T., Getz, G., Hadley, K., … Montgomery, S. B. (2017). Genetic effects on gene expression across human tissues. Nature, 550(7675), 204–213. <a href="https://doi.org/10.1038/nature24277">https://doi.org/10.1038/nature24277</a></li>
    </ul>
  </li>
  <li>Geuvadis study
    <ul>
      <li>Lappalainen, T., Sammeth, M., Friedländer, M. R., ’T Hoen, P. A. C., Monlong, J., Rivas, M. A., … Dermitzakis, E. T. (2013). Transcriptome and genome sequencing uncovers functional variation in humans. Nature, 501(7468), 506–511. <a href="https://doi.org/10.1038/nature12531">https://doi.org/10.1038/nature12531</a></li>
    </ul>
  </li>
  <li><a href="https://www.cell.com/consortium/IHEC">IHEC</a>
    <ul>
      <li>Chen, L., Ge, B., Casale, F. P., Vasquez, L., Kwan, T., Garrido-Martín, D., … Soranzo, N. (2016). Genetic Drivers of Epigenetic and Transcriptional Variation in Human Immune Cells. Cell (Vol. 167). <a href="https://doi.org/10.1016/j.cell.2016.10.026">https://doi.org/10.1016/j.cell.2016.10.026</a></li>
    </ul>
  </li>
</ul>

<h4 id="splicing-qtl">splicing QTL</h4>

<ul>
  <li>Li, Y. I., van de Geijn, B., Raj, A., Knowles, D. a, Petti, A. a, Golan, D., … Pritchard, J. K. (2016). RNA splicing is a primary link between genetic variation and disease. Science, 352(6285), 600–604. <a href="https://doi.org/10.1126/science.aad9417">https://doi.org/10.1126/science.aad9417</a></li>
  <li>Li, Y. I., van de Geijn, B., Raj, A., Knowles, D. a, Petti, A. a, Golan, D., … Pritchard, J. K. (2016). RNA splicing is a primary link between genetic variation and disease. Science, 352(6285), 600–604. <a href="https://doi.org/10.1126/science.aad9417">https://doi.org/10.1126/science.aad9417</a></li>
</ul>

<h4 id="ctqtl-cell-type-specific">ctQTL (cell-type-specific)</h4>

<ul>
  <li>Ishigaki, K., Kochi, Y., Suzuki, A., Tsuchida, Y., Tsuchiya, H., Sumitomo, S., … Yamamoto, K. (2017). Polygenic burdens on cell-specific pathways underlie the risk of rheumatoid arthritis. Nature Genetics, 49(7), 1120–1125. <a href="https://doi.org/10.1038/ng.3885">https://doi.org/10.1038/ng.3885</a></li>
</ul>

<h4 id="scqtl-single-cell">scQTL (single-cell)</h4>

<ul>
  <li>Ishigaki, K., Kochi, Y., Suzuki, A., Tsuchida, Y., Tsuchiya, H., Sumitomo, S., … Yamamoto, K. (2017). Polygenic burdens on cell-specific pathways underlie the risk of rheumatoid arthritis. Nature Genetics, 49(7), 1120–1125. <a href="https://doi.org/10.1038/ng.3885">https://doi.org/10.1038/ng.3885</a></li>
</ul>

<h4 id="allelic-specific-exprssion-qtl">allelic specific exprssion QTL</h4>

<h3 id="pqtl-protein">pQTL (protein)</h3>

<ul>
  <li><a href="http://www.intervalstudy.org.uk/">INTERVAL study</a></li>
  <li>Sun, B. B., Maranville, J. C., Peters, J. E., Stacey, D., Staley, J. R., Blackshaw, J., … Butterworth, A. S. (2018). Genomic atlas of the human plasma proteome. Nature, 558(7708), 73–79. <a href="https://doi.org/10.1038/s41586-018-0175-2">https://doi.org/10.1038/s41586-018-0175-2</a></li>
</ul>


  </div>

</article>

      </div>
    </div>

    <footer class="site-footer">

  <div class="wrapper">


    <div class="footer-col-wrapper">
      <div class="footer-col footer-col-1">
        <ul class="contact-list">
          <li>Huanwei Wang's blog</li>
          <li><a href="mailto:zju_whw@163.com">zju_whw@163.com</a></li>
        </ul>
      </div>

      <div class="footer-col footer-col-2">
        <ul class="social-media-list">
          
          <li>
            <a href="https://github.com/zjuwhw"><span class="icon icon--github"><svg viewBox="0 0 16 16"><path fill="#828282" d="M7.999,0.431c-4.285,0-7.76,3.474-7.76,7.761 c0,3.428,2.223,6.337,5.307,7.363c0.388,0.071,0.53-0.168,0.53-0.374c0-0.184-0.007-0.672-0.01-1.32 c-2.159,0.469-2.614-1.04-2.614-1.04c-0.353-0.896-0.862-1.135-0.862-1.135c-0.705-0.481,0.053-0.472,0.053-0.472 c0.779,0.055,1.189,0.8,1.189,0.8c0.692,1.186,1.816,0.843,2.258,0.645c0.071-0.502,0.271-0.843,0.493-1.037 C4.86,11.425,3.049,10.76,3.049,7.786c0-0.847,0.302-1.54,0.799-2.082C3.768,5.507,3.501,4.718,3.924,3.65 c0,0,0.652-0.209,2.134,0.796C6.677,4.273,7.34,4.187,8,4.184c0.659,0.003,1.323,0.089,1.943,0.261 c1.482-1.004,2.132-0.796,2.132-0.796c0.423,1.068,0.157,1.857,0.077,2.054c0.497,0.542,0.798,1.235,0.798,2.082 c0,2.981-1.814,3.637-3.543,3.829c0.279,0.24,0.527,0.713,0.527,1.437c0,1.037-0.01,1.874-0.01,2.129 c0,0.208,0.14,0.449,0.534,0.373c3.081-1.028,5.302-3.935,5.302-7.362C15.76,3.906,12.285,0.431,7.999,0.431z"/></svg>
</span><span class="username">zjuwhw</span></a>

          </li>
          

          
          <li>
            <a href="https://twitter.com/huanwei_wang"><span class="icon icon--twitter"><svg viewBox="0 0 16 16"><path fill="#828282" d="M15.969,3.058c-0.586,0.26-1.217,0.436-1.878,0.515c0.675-0.405,1.194-1.045,1.438-1.809c-0.632,0.375-1.332,0.647-2.076,0.793c-0.596-0.636-1.446-1.033-2.387-1.033c-1.806,0-3.27,1.464-3.27,3.27 c0,0.256,0.029,0.506,0.085,0.745C5.163,5.404,2.753,4.102,1.14,2.124C0.859,2.607,0.698,3.168,0.698,3.767 c0,1.134,0.577,2.135,1.455,2.722C1.616,6.472,1.112,6.325,0.671,6.08c0,0.014,0,0.027,0,0.041c0,1.584,1.127,2.906,2.623,3.206 C3.02,9.402,2.731,9.442,2.433,9.442c-0.211,0-0.416-0.021-0.615-0.059c0.416,1.299,1.624,2.245,3.055,2.271 c-1.119,0.877-2.529,1.4-4.061,1.4c-0.264,0-0.524-0.015-0.78-0.046c1.447,0.928,3.166,1.469,5.013,1.469 c6.015,0,9.304-4.983,9.304-9.304c0-0.142-0.003-0.283-0.009-0.423C14.976,4.29,15.531,3.714,15.969,3.058z"/></svg>
</span><span class="username">huanwei_wang</span></a>

          </li>
          
        </ul>
      </div>

      <div class="footer-col footer-col-3">
        <p>This is Huanwei Wang's personal blog. Powered by Jekyll and Host in Github.
</p>
      </div>
    </div>

  </div>

</footer>


  </body>

</html>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    processEscapes: true
  }
});
</script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js?config=TeX-AMS-MML_HTMLorMML" type="text/javascript"></script>
