<!DOCTYPE html>
<html>

  <head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  <title>HWE and LD</title>
  <meta name="description" content="Hardy-Weinberg equilibrium and Linkage disequilibrium are two basic concepts in population genetics.">

  <link rel="stylesheet" href="/css/main.css">
  <link rel="canonical" href="http://localhost:4000/2017/06/07/HWE_LD.html">
  <link rel="alternate" type="application/rss+xml" title="Huanwei Wang's blog" href="http://localhost:4000/feed.xml">
</head>


  <body>

    <!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-121065275-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-121065275-1');
</script>



    <header class="site-header">

  <div class="wrapper">

    <a class="site-title" href="/">Huanwei Wang's blog</a>

    <nav class="site-nav">
      <a href="#" class="menu-icon">
        <svg viewBox="0 0 18 15">
          <path fill="#424242" d="M18,1.484c0,0.82-0.665,1.484-1.484,1.484H1.484C0.665,2.969,0,2.304,0,1.484l0,0C0,0.665,0.665,0,1.484,0 h15.031C17.335,0,18,0.665,18,1.484L18,1.484z"/>
          <path fill="#424242" d="M18,7.516C18,8.335,17.335,9,16.516,9H1.484C0.665,9,0,8.335,0,7.516l0,0c0-0.82,0.665-1.484,1.484-1.484 h15.031C17.335,6.031,18,6.696,18,7.516L18,7.516z"/>
          <path fill="#424242" d="M18,13.516C18,14.335,17.335,15,16.516,15H1.484C0.665,15,0,14.335,0,13.516l0,0 c0-0.82,0.665-1.484,1.484-1.484h15.031C17.335,12.031,18,12.696,18,13.516L18,13.516z"/>
        </svg>
      </a>

      <div class="trigger">
        
          
        
          
          <a class="page-link" href="/about/">About</a>
          
        
          
        
          
        
          
        
          
        
          
        
          
          <a class="page-link" href="/links/">Links</a>
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
          <a class="page-link" href="/tags/">Tags</a>
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
      </div>
    </nav>

  </div>

</header>


    <div class="page-content">
      <div class="wrapper">
        <article class="post" itemscope itemtype="http://schema.org/BlogPosting">

  <header class="post-header">
    <h1 class="post-title" itemprop="name headline">HWE and LD</h1>
    <p class="post-meta"><time datetime="2017-06-07T00:00:00+10:00" itemprop="datePublished">Jun 7, 2017</time></p>
  </header>

  <div class="post-content" itemprop="articleBody">
    <p><a href="https://en.wikipedia.org/wiki/Hardy%E2%80%93Weinberg_principle">Hardy-Weinberg equilibrium</a> and <a href="https://en.wikipedia.org/wiki/Linkage_disequilibrium">Linkage disequilibrium</a> are two basic concepts in <a href="https://en.wikipedia.org/wiki/Population_genetics">population genetics</a>.</p>

<h3 id="defination">Defination</h3>

<ul>
  <li>Hardy-Weinberg Equilibrium (HWE): <strong>allele and genotype frequencies</strong> in a population will remain constant <strong>from generation to generation</strong> in the absence of other evolutionary influences.</li>
  <li>Linkage Disequilibrium (LD): the non-random association of <strong>alleles at different loci</strong> in a given population.</li>
</ul>

<p>So HWE is about the allele and genotype frequencies in one locus, but LD is about alleles at different loci.</p>

<h3 id="hardy-weinberg-equilibrium">Hardy-Weinberg Equilibrium</h3>

<h4 id="from-genotype-to-allele-frequencies">From Genotype to Allele frequencies</h4>

<p>For $n_{AA}, n_{Aa}, n_{aa}$ is the count of genotype AA, Aa and aa, respectively. And $n = n_{AA} + n_{Aa} + n_{aa}$.</p>

<p>So, the count of A and a alleles are:</p>

\[n_{A} = 2 \times n_{AA} + n_{Aa}\]

\[n_{a} = 2 \times n_{aa} + n_{Aa}\]

<p>The frequencies of A and a alleles are:</p>

\[p_{A} = \frac{n_{A}}{2n} = \frac{2 \times n_{AA} + n_{Aa}}{2n} = p_{AA} + \frac{1}{2}p_{Aa}\]

\[p_{a} = \frac{n_{a}}{2n} = \frac{2 \times n_{aa} + n_{Aa}}{2n} = p_{aa} + \frac{1}{2}p_{Aa}\]

\[p_{A} = 1 - p_{a}\]

<h4 id="from-allele-to-genotype-frequencies---hwe">From Allele to Genotype frequencies - HWE</h4>

<p>We cannot directly get genotype frequencies from allele frequencies. But under Hardy-Weiberge Equilibrium, we can do it by</p>

\[p_{AA} = p_{A}^{2}\]

\[p_{Aa} = 2p_{A}p_{a}\]

\[p_{aa} = p_{a}^{2}\]

<p>The assumptions for HWE includes:</p>

<ul>
  <li>random mating</li>
  <li>infinite population size (no genetic drift)</li>
  <li>no mutation</li>
  <li>no selection</li>
  <li>no migration</li>
</ul>

<p>In addition, we can see genotype frequencies is a binomial distribution. If SNP genotpes are coded X = 0, 1 and 2 (alleles) and the allele frequency is p, then:</p>

\[X \sim B(2, p)\text{ with }E(X) = 2p\text{ and }Var(X)=2p(1-p)\]

<p>We can plot the distribution of three different genotypes. In real genetics, we usually use the minor allele, so p is less than 0.5.</p>

<pre><code class="language-{r}">p = seq(0,1,0.01)
plot(p, (1-p)^2, ylab = "genotype frequency", main="Under HWE",type="l", col="blue")
lines(p, 2*p*(1-p), col="green")
lines(p, p^2, col="grey")
legend(0.5,0.95,"genotype = 0",lty = 1, col ="blue", bty="n")
legend(0.5,0.85,"genotype = 1",lty = 1, col ="green", bty = "n")
legend(0.5,0.75,"genotype = 2",lty = 1, col ="grey", bty = "n")
abline(v = 0.5, lty=2)
</code></pre>

<p><img src="/images/HWE_allele_genotype.png" alt="" /></p>

<h4 id="test-for-hwe">Test for HWE</h4>

<p>In the GWAS QC step, we usually remove SNPs with extensive deviation from HWE because this can be indicative of a genotyping or genotype-calling error.</p>

<p>The two common ways to test HWE are Chi-square test and exact test.</p>

<p>For Chi-square test:</p>

\[\chi^{2} = \frac{(n_{AA}-e{AA})^2}{e_{AA}} + \frac{(n_{Aa}-e{Aa})^2}{e_{Aa}} + \frac{(n_{aa}-e{aa})^2}{e_{aa}}\]

<p>where $e_{AA} = np_{A}^2$, $e_{Aa} = 2np_{A}p_{a}$, $e_{aa} = np_{a}^2$ and df = 1 (#genotype - #allele).</p>

<p>For exact test:</p>

\[P(N_{Aa}|N_{A}) = \frac{n_{A}!n_{a}!n!2^{n_{Aa}}}{\frac{1}{2}(n_{A}-n_{Aa})!n_{Aa}!\frac{1}{2}(n_{a}-n_{Aa})!(2n)!}\]

<p>For example, a sample with 2504 individuals, the observed genotypes are 2467/36/1 for AA, Aa and aa. We can do the chi-square test in R.</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">### Observed genotype counts</span><span class="w">
</span><span class="n">n_AA</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2467</span><span class="w">
</span><span class="n">n_Aa</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">36</span><span class="w">
</span><span class="n">n_aa</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="w">
</span><span class="n">n</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">n_AA</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">n_Aa</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">n_aa</span><span class="w">
</span><span class="n">O_geno</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="n">n_AA</span><span class="p">,</span><span class="w"> </span><span class="n">n_Aa</span><span class="p">,</span><span class="w"> </span><span class="n">n_aa</span><span class="p">)</span><span class="w">
</span><span class="n">cat</span><span class="p">(</span><span class="n">O_geno</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 2467 36 1
</code></pre></div></div>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">### Observed allele count and frequency</span><span class="w">
</span><span class="n">n_A</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">n_AA</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">n_Aa</span><span class="w">
</span><span class="n">n_a</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">n_aa</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">n_Aa</span><span class="w">
</span><span class="n">p_A</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">n_A</span><span class="o">/</span><span class="p">(</span><span class="m">2</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="n">n</span><span class="p">)</span><span class="w">
</span><span class="n">p_a</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">n_a</span><span class="o">/</span><span class="p">(</span><span class="m">2</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="n">n</span><span class="p">)</span><span class="w">
</span><span class="n">cat</span><span class="p">(</span><span class="n">p_A</span><span class="p">,</span><span class="w"> </span><span class="n">p_a</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 0.9924121 0.007587859
</code></pre></div></div>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">### Expected genotype count, under HWE</span><span class="w">
</span><span class="n">E_AA</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">p_A</span><span class="o">^</span><span class="m">2</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="n">n</span><span class="w">
</span><span class="n">E_Aa</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="n">p_A</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="n">p_a</span><span class="w"> </span><span class="o">*</span><span class="w">  </span><span class="n">n</span><span class="w">
</span><span class="n">E_aa</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">p_a</span><span class="o">^</span><span class="m">2</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="n">n</span><span class="w">
</span><span class="n">E_geno</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="n">E_AA</span><span class="p">,</span><span class="w"> </span><span class="n">E_Aa</span><span class="p">,</span><span class="w"> </span><span class="n">E_aa</span><span class="p">)</span><span class="w">
</span><span class="n">cat</span><span class="p">(</span><span class="n">E_geno</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 2466.144 37.71166 0.1441693
</code></pre></div></div>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">### Chi-square with df=1 (#genotypes - #alleles)</span><span class="w">
</span><span class="n">chisq</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">sum</span><span class="p">((</span><span class="n">O_geno</span><span class="o">-</span><span class="n">E_geno</span><span class="p">)</span><span class="o">^</span><span class="m">2</span><span class="o">/</span><span class="n">E_geno</span><span class="p">)</span><span class="w">
</span><span class="n">pvalue</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="n">pchisq</span><span class="p">(</span><span class="n">chisq</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">)</span><span class="w">
</span><span class="n">cat</span><span class="p">(</span><span class="n">paste</span><span class="p">(</span><span class="s2">"chisq:"</span><span class="p">,</span><span class="w"> </span><span class="n">chisq</span><span class="p">,</span><span class="w"> </span><span class="s2">";"</span><span class="p">,</span><span class="s2">"pvalue: "</span><span class="p">,</span><span class="w"> </span><span class="n">pvalue</span><span class="p">))</span><span class="w">
</span></code></pre></div></div>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## chisq: 5.15844350552391 ; pvalue:  0.02313362043297
</code></pre></div></div>

<p>We can also use the R package “HardyWeinberg” to do the chi-square test.</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">### We can also use the R package "HardyWeinberg"</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">HardyWeinberg</span><span class="p">)</span><span class="w">
</span><span class="n">HWChisq</span><span class="p">(</span><span class="n">O_geno</span><span class="p">,</span><span class="w"> </span><span class="n">cc</span><span class="o">=</span><span class="m">0</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## Chi-square test for Hardy-Weinberg equilibrium (autosomal)
## Chi2 =  5.158444 DF =  1 p-value =  0.02313362 D =  -0.8558307 f =  0.04538812
</code></pre></div></div>

<p>It’s hard to do the exact test directly in R because of the factorial calucation. So we also use the R package “HardyWeinberg”. A more common way is to use <a href="https://www.cog-genomics.org/plink/1.9/filter#hwe">PLINK</a> “–hwe” option.</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">HWExact</span><span class="p">(</span><span class="n">O_geno</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## Haldane Exact test for Hardy-Weinberg equilibrium (autosomal)
## using SELOME p-value
## sample counts: n =  2467 n =  36 n =  1 
## H0: HWE (D==0), H1: D &lt;&gt; 0 
## D =  -0.8558307 p-value =  0.1318891
</code></pre></div></div>

<p>We can see that the p-values from chi-square and exact test are quite different. In 2005, <a href="http://www.sciencedirect.com/science/article/pii/S0002929707607356">Wigginton et al</a> has shown that the chi-square inflated type I error rates. So exact test is a better method.</p>

<h3 id="linkage-disequilibrium">Linkage disequilibrium</h3>

<h4 id="disequilibrium-coefficient-d">Disequilibrium coefficient (D)</h4>

<p>For a pair of biallelic loci, there are four <a href="https://en.wikipedia.org/wiki/Haplotype">haplotypes</a>. If there is no association between these two SNPs (linkage equilibrium), the haplotype frequencies should be:</p>

<ul>
  <li>$p_{AB} = p_{A} \times p_{B}$</li>
  <li>$p_{aB} = p_{a} \times p_{B}$</li>
  <li>$p_{Ab} = p_{A} \times p_{b}$</li>
  <li>$p_{ab} = p_{a} \times p_{b}$</li>
</ul>

<p>However, if the two SNPs have nonrandom association, the LD can be defined by disequilibrium coefficient:</p>

<ul>
  <li>$D_{AB} = p_{AB} - p_{A} \times p_{B}$</li>
  <li>$P_{AB} = p_{A} \times p_{B} + D_{AB}$</li>
  <li>$p_{aB} = p_{a} \times p_{B} - D_{AB}$</li>
  <li>$p_{Ab} = p_{A} \times p_{b} - D_{AB}$</li>
  <li>$p_{ab} = p_{a} \times p_{b} + D_{AB}$</li>
</ul>

<h4 id="other-ld-measures">Other LD measures</h4>

<p>More wildely used measures for LD are D’ (LEWONTIN 1964) and $r^{2}$ (HILL and ROBERTSON 1968).</p>

\[D'_{AB} = \frac{D_{AB}}{D_{min}}\]

\[D_{min} = max(-p_{A}p_{B}, -p_{a}p_{b}), if D_{AB} &lt; 0\]

\[D_{min} = min(p_{A}p_{b}, p_{a}p_{B}), if D_{AB} &gt; 0\]

<p>D′ = 1 indicates that at least one of the four possible haplotypes is absent, regardless of the allele frequencies.</p>

\[r^{2} = \frac{D^{2}}{p_{A}p_{a}p_{B}p_{b}}\]

<p>$r^{2}$ is a correlation coefficient of 1/0 (all or none) indicator variables indicating the presence of A and B.</p>

<h4 id="influence-factor">Influence Factor</h4>

<p>Linkage disequilibrium is influenced by many factors, including,</p>
<ul>
  <li>selection</li>
  <li>the rate of recombination</li>
  <li>the rate of mutation</li>
  <li>genetic drift</li>
  <li>the system of mating</li>
  <li>population structure</li>
  <li>genetic linkage.</li>
</ul>

<h4 id="an-example-in-r">An example in R</h4>

<p>Haplotype table</p>

<table>
  <thead>
    <tr>
      <th style="text-align: left">haplotype</th>
      <th style="text-align: left">ab</th>
      <th style="text-align: left">Ab</th>
      <th style="text-align: left">aB</th>
      <th style="text-align: left">AB</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">frequency</td>
      <td style="text-align: left">0</td>
      <td style="text-align: left">32/5008</td>
      <td style="text-align: left">1/5008</td>
      <td style="text-align: left">4975/5008</td>
    </tr>
  </tbody>
</table>

<p>2*2 allele frequency table</p>

<table>
  <thead>
    <tr>
      <th style="text-align: left">allele</th>
      <th style="text-align: left">a</th>
      <th style="text-align: left">A</th>
      <th style="text-align: left">Total</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">b</td>
      <td style="text-align: left">0</td>
      <td style="text-align: left">32/5008</td>
      <td style="text-align: left">32/5008</td>
    </tr>
    <tr>
      <td style="text-align: left">B</td>
      <td style="text-align: left">1/5008</td>
      <td style="text-align: left">4975/5008</td>
      <td style="text-align: left">4976/5008</td>
    </tr>
    <tr>
      <td style="text-align: left">Total</td>
      <td style="text-align: left">1/5008</td>
      <td style="text-align: left">5007/5008</td>
      <td style="text-align: left">1</td>
    </tr>
  </tbody>
</table>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">#haplotype frequency</span><span class="w">
</span><span class="n">pab</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0</span><span class="w">
</span><span class="n">pAb</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">32</span><span class="o">/</span><span class="m">5008</span><span class="w">
</span><span class="n">paB</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="o">/</span><span class="m">5008</span><span class="w">
</span><span class="n">pAB</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">4975</span><span class="o">/</span><span class="m">5008</span><span class="w">
</span><span class="n">cat</span><span class="p">(</span><span class="n">pab</span><span class="p">,</span><span class="w"> </span><span class="n">pAb</span><span class="p">,</span><span class="w"> </span><span class="n">paB</span><span class="p">,</span><span class="w"> </span><span class="n">pAB</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 0 0.006389776 0.0001996805 0.9934105
</code></pre></div></div>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">#allele frequency</span><span class="w">
</span><span class="n">pa</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">paB</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">pab</span><span class="w">
</span><span class="n">pA</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">pAB</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">pAb</span><span class="w">
</span><span class="n">pb</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">pab</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">pAb</span><span class="w">
</span><span class="n">pB</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">paB</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">pAB</span><span class="w">
</span><span class="n">cat</span><span class="p">(</span><span class="n">pa</span><span class="p">,</span><span class="w"> </span><span class="n">pA</span><span class="p">,</span><span class="w"> </span><span class="n">pb</span><span class="p">,</span><span class="w"> </span><span class="n">pB</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 0.0001996805 0.9998003 0.006389776 0.9936102
</code></pre></div></div>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">#D</span><span class="w">
</span><span class="n">D</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">pAB</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="n">pA</span><span class="o">*</span><span class="n">pB</span><span class="w">
</span><span class="n">cat</span><span class="p">(</span><span class="n">D</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## -1.275914e-06
</code></pre></div></div>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">#Dmin</span><span class="w">
</span><span class="n">Dmin</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">max</span><span class="p">(</span><span class="o">-</span><span class="n">pA</span><span class="o">*</span><span class="n">pB</span><span class="p">,</span><span class="w"> </span><span class="o">-</span><span class="n">pa</span><span class="o">*</span><span class="n">pb</span><span class="p">)</span><span class="w">
</span><span class="n">cat</span><span class="p">(</span><span class="n">Dmin</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## -1.275914e-06
</code></pre></div></div>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">#D'</span><span class="w">
</span><span class="n">Ddot</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">D</span><span class="o">/</span><span class="n">Dmin</span><span class="w">
</span><span class="n">cat</span><span class="p">(</span><span class="n">Ddot</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 1
</code></pre></div></div>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">#r-square</span><span class="w">
</span><span class="n">rsq</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">D</span><span class="o">^</span><span class="m">2</span><span class="o">/</span><span class="p">(</span><span class="n">pa</span><span class="o">*</span><span class="n">pA</span><span class="o">*</span><span class="n">pb</span><span class="o">*</span><span class="n">pB</span><span class="p">)</span><span class="w">
</span><span class="n">cat</span><span class="p">(</span><span class="n">rsq</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 1.284376e-06
</code></pre></div></div>

<h3 id="reference">Reference</h3>

<ol>
  <li><a href="http://bioinformatics.org.au/ws09/presentations/Day3_PVisscher.pdf">ppt from Peter Visscher</a></li>
  <li><a href="http://www.springer.com/cn/book/9781441973375">Laird et al, The Fundamentals of Modern Statistical Genetics, 2011</a></li>
  <li><a href="https://cran.r-project.org/web/packages/HardyWeinberg/vignettes/HardyWeinberg.pdf">R package HardyWeinberg</a></li>
  <li><a href="https://www.ncbi.nlm.nih.gov/pubmed/17248194">Lewontin RC. The Interaction of Selection and Linkage. I. General Considerations; Heterotic Models. Genetics, 1964.</a></li>
  <li><a href="https://link.springer.com/article/10.1007/BF01245622">Hill W.G. Linkage disequilibrium in finite populations. Theoretical and Applied Genetics, 1968.</a></li>
  <li><a href="http://csg.sph.umich.edu/abecasis/class/">Gonçalo’s Lecture Notes - Models in Human Genetics</a></li>
</ol>

  </div>

</article>

      </div>
    </div>

    <footer class="site-footer">

  <div class="wrapper">


    <div class="footer-col-wrapper">
      <div class="footer-col footer-col-1">
        <ul class="contact-list">
          <li>Huanwei Wang's blog</li>
          <li><a href="mailto:zju_whw@163.com">zju_whw@163.com</a></li>
        </ul>
      </div>

      <div class="footer-col footer-col-2">
        <ul class="social-media-list">
          
          <li>
            <a href="https://github.com/zjuwhw"><span class="icon icon--github"><svg viewBox="0 0 16 16"><path fill="#828282" d="M7.999,0.431c-4.285,0-7.76,3.474-7.76,7.761 c0,3.428,2.223,6.337,5.307,7.363c0.388,0.071,0.53-0.168,0.53-0.374c0-0.184-0.007-0.672-0.01-1.32 c-2.159,0.469-2.614-1.04-2.614-1.04c-0.353-0.896-0.862-1.135-0.862-1.135c-0.705-0.481,0.053-0.472,0.053-0.472 c0.779,0.055,1.189,0.8,1.189,0.8c0.692,1.186,1.816,0.843,2.258,0.645c0.071-0.502,0.271-0.843,0.493-1.037 C4.86,11.425,3.049,10.76,3.049,7.786c0-0.847,0.302-1.54,0.799-2.082C3.768,5.507,3.501,4.718,3.924,3.65 c0,0,0.652-0.209,2.134,0.796C6.677,4.273,7.34,4.187,8,4.184c0.659,0.003,1.323,0.089,1.943,0.261 c1.482-1.004,2.132-0.796,2.132-0.796c0.423,1.068,0.157,1.857,0.077,2.054c0.497,0.542,0.798,1.235,0.798,2.082 c0,2.981-1.814,3.637-3.543,3.829c0.279,0.24,0.527,0.713,0.527,1.437c0,1.037-0.01,1.874-0.01,2.129 c0,0.208,0.14,0.449,0.534,0.373c3.081-1.028,5.302-3.935,5.302-7.362C15.76,3.906,12.285,0.431,7.999,0.431z"/></svg>
</span><span class="username">zjuwhw</span></a>

          </li>
          

          
          <li>
            <a href="https://twitter.com/huanwei_wang"><span class="icon icon--twitter"><svg viewBox="0 0 16 16"><path fill="#828282" d="M15.969,3.058c-0.586,0.26-1.217,0.436-1.878,0.515c0.675-0.405,1.194-1.045,1.438-1.809c-0.632,0.375-1.332,0.647-2.076,0.793c-0.596-0.636-1.446-1.033-2.387-1.033c-1.806,0-3.27,1.464-3.27,3.27 c0,0.256,0.029,0.506,0.085,0.745C5.163,5.404,2.753,4.102,1.14,2.124C0.859,2.607,0.698,3.168,0.698,3.767 c0,1.134,0.577,2.135,1.455,2.722C1.616,6.472,1.112,6.325,0.671,6.08c0,0.014,0,0.027,0,0.041c0,1.584,1.127,2.906,2.623,3.206 C3.02,9.402,2.731,9.442,2.433,9.442c-0.211,0-0.416-0.021-0.615-0.059c0.416,1.299,1.624,2.245,3.055,2.271 c-1.119,0.877-2.529,1.4-4.061,1.4c-0.264,0-0.524-0.015-0.78-0.046c1.447,0.928,3.166,1.469,5.013,1.469 c6.015,0,9.304-4.983,9.304-9.304c0-0.142-0.003-0.283-0.009-0.423C14.976,4.29,15.531,3.714,15.969,3.058z"/></svg>
</span><span class="username">huanwei_wang</span></a>

          </li>
          
        </ul>
      </div>

      <div class="footer-col footer-col-3">
        <p>This is Huanwei Wang's personal blog. Powered by Jekyll and Host in Github.
</p>
      </div>
    </div>

  </div>

</footer>


  </body>

</html>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    processEscapes: true
  }
});
</script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js?config=TeX-AMS-MML_HTMLorMML" type="text/javascript"></script>
