<!DOCTYPE html>
<html>

  <head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  <title>Plink bfile manipulation</title>
  <meta name="description" content="Plink is one of the most widely used software in GWAS field and its relative file format bfile, a binary file used to store genotype information, may be more...">

  <link rel="stylesheet" href="/css/main.css">
  <link rel="canonical" href="http://localhost:4000/2017/08/18/plink_bfile.html">
  <link rel="alternate" type="application/rss+xml" title="Huanwei Wang's blog" href="http://localhost:4000/feed.xml">
</head>


  <body>

    <!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-121065275-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-121065275-1');
</script>



    <header class="site-header">

  <div class="wrapper">

    <a class="site-title" href="/">Huanwei Wang's blog</a>

    <nav class="site-nav">
      <a href="#" class="menu-icon">
        <svg viewBox="0 0 18 15">
          <path fill="#424242" d="M18,1.484c0,0.82-0.665,1.484-1.484,1.484H1.484C0.665,2.969,0,2.304,0,1.484l0,0C0,0.665,0.665,0,1.484,0 h15.031C17.335,0,18,0.665,18,1.484L18,1.484z"/>
          <path fill="#424242" d="M18,7.516C18,8.335,17.335,9,16.516,9H1.484C0.665,9,0,8.335,0,7.516l0,0c0-0.82,0.665-1.484,1.484-1.484 h15.031C17.335,6.031,18,6.696,18,7.516L18,7.516z"/>
          <path fill="#424242" d="M18,13.516C18,14.335,17.335,15,16.516,15H1.484C0.665,15,0,14.335,0,13.516l0,0 c0-0.82,0.665-1.484,1.484-1.484h15.031C17.335,12.031,18,12.696,18,13.516L18,13.516z"/>
        </svg>
      </a>

      <div class="trigger">
        
          
        
          
          <a class="page-link" href="/about/">About</a>
          
        
          
        
          
        
          
        
          
        
          
        
          
          <a class="page-link" href="/links/">Links</a>
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
          <a class="page-link" href="/tags/">Tags</a>
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
      </div>
    </nav>

  </div>

</header>


    <div class="page-content">
      <div class="wrapper">
        <article class="post" itemscope itemtype="http://schema.org/BlogPosting">

  <header class="post-header">
    <h1 class="post-title" itemprop="name headline">Plink bfile manipulation</h1>
    <p class="post-meta"><time datetime="2017-08-18T00:00:00+10:00" itemprop="datePublished">Aug 18, 2017</time></p>
  </header>

  <div class="post-content" itemprop="articleBody">
    <p><a href="https://www.cog-genomics.org/plink2">Plink</a> is one of the most widely used software in GWAS field and its relative file format <a href="https://www.cog-genomics.org/plink/1.9/formats#bed">bfile</a>, a binary file used to store genotype information, may be more popular as supported by other softwares (e.g. <a href="http://cnsgenomics.com/software/gcta/#Overview">GCTA</a>) as input files.</p>

<h2 id="pedmap-files">“ped/map” files</h2>

<p>In the plink file formats, <code class="language-plaintext highlighter-rouge">.ped/map</code> files are text files used together to store pedigree + genotype information and variant information, respectively. For example, the below commands show the basic information of the <code class="language-plaintext highlighter-rouge">test.map</code> and <code class="language-plaintext highlighter-rouge">test.ped</code>, which is converted from test.bed/bim/fam files in gcta 1.26 software package.</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>$ head -2 test.map
1	rs4475691	0	836671
1	rs28705211	0.0151734	890368

$ wc -l test.map
    1000 test.map
    
$ head -2 test.ped |cut -f 1-8 -d " "
1 11 0 0 1 -9 C C
2 21 0 0 2 -9 C C

$ wc -l test.ped
    3925 test.ped
$ head -1 test.ped|awk '{print NF}'
2006
</code></pre></div></div>

<p><code class="language-plaintext highlighter-rouge">test.map</code> has 1000 lines, which represent 1000 variants with chromosome, SNP id, position in morgans and position in BP per line.</p>

<p><code class="language-plaintext highlighter-rouge">test.ped</code> has 3925 lines, which represent 3925 individuals with pedigree information in the first 6 columns and genotype information for the other columns. As one genotype has two alleles, <code class="language-plaintext highlighter-rouge">test.ped</code> has 6 + 2 * 1000 = 2006 columns.</p>

<h2 id="bedbimfam-files">“bed/bim/fam” files</h2>

<p>The <code class="language-plaintext highlighter-rouge">.bed/bim/fam</code> files are also called bfile, which is the binary file of <code class="language-plaintext highlighter-rouge">.ped/map</code> files. The bed file is the test file for genotype, the bim file is for variant information (similiar with map file, added with reference and alternative allele) and the fam file is the text file for pedigree information (first 6 columns of ped file).</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>$ head -2 test.bim
1	rs4475691	0	836671	T	C
1	rs28705211	0.0151734	890368	C	G
$ head -2 test.fam
1 11 0 0 1 -9
2 21 0 0 2 -9
</code></pre></div></div>

<p>The <code class="language-plaintext highlighter-rouge">.bed/bim/fam</code> files are easy to read/write for softwares and can reduce the storage by more than 10-fold (30840+15776286/982003+34840+76286).</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>$ ls -l test.ped test.map
-rw-r--r--  1 huanwei.wang  Group-Yang     30840 18 Aug 13:42 test.map
-rw-r--r--  1 huanwei.wang  Group-Yang  15776286 18 Aug 13:42 test.ped
$ ls -l test.bed test.bim test.fam
-rw-r-----  1 huanwei.wang  Group-Yang  982003 18 Aug 10:44 test.bed
-rw-r-----  1 huanwei.wang  Group-Yang   34840 18 Aug 10:44 test.bim
-rw-r-----  1 huanwei.wang  Group-Yang   76286 18 Aug 10:44 test.fam
$ echo ""|awk '{print (30840+15776286)/(982003+34840+76286)}'
14.4604
</code></pre></div></div>

<p>The <code class="language-plaintext highlighter-rouge">.bed/bim/fam</code> and <code class="language-plaintext highlighter-rouge">.ped/map</code> files can be converted to each other by plink software.</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code># bfile to ped/map
$ plink2 --bfile test --recode --out test

# ped/map to bfile
$ plink2 --file test --make-bed --out test
</code></pre></div></div>

<h2 id="r-package-plink2r">R package “plink2R”</h2>

<p>Sometimes we need to use bfile to do more statsitcial analysis in R, so we need to read the bfile. One strategy is to convert it into text files <code class="language-plaintext highlighter-rouge">ped/map</code>, while we can also use R package <a href="https://github.com/gabraham/plink2R">plink2R</a> to read directly.</p>

<p>Installation</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>devtools::install_github("gabraham/plink2R/plink2R")
</code></pre></div></div>

<p>Usage</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>&gt; library(plink2R)
Loading required package: Rcpp
Loading required package: RcppEigen
Warning message:
package ‘Rcpp’ was built under R version 3.4.1
&gt; dat &lt;- read_plink("test")
&gt; dim(dat$bed)
[1] 3925 1000
&gt; dim(dat$fam)
[1] 3925    6
&gt; dim(dat$bim)
[1] 1000    6
</code></pre></div></div>

<p>But the plink2R is relatively slow and takes more memory, so try to manipulate bfile using plink as much as possible.</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>&gt; object.size(dat)
31908168 bytes
</code></pre></div></div>

<h2 id="c-ad-python-library-plinkio">C ad Python library “plinkio”</h2>

<p>There is also packages for C and Python lauguage, e.g. <a href="https://github.com/mfranberg/libplinkio">plinkio</a>.</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>from plinkio import plinkfile

plink_file = plinkfile.open( "test" )
</code></pre></div></div>

  </div>

</article>

      </div>
    </div>

    <footer class="site-footer">

  <div class="wrapper">


    <div class="footer-col-wrapper">
      <div class="footer-col footer-col-1">
        <ul class="contact-list">
          <li>Huanwei Wang's blog</li>
          <li><a href="mailto:zju_whw@163.com">zju_whw@163.com</a></li>
        </ul>
      </div>

      <div class="footer-col footer-col-2">
        <ul class="social-media-list">
          
          <li>
            <a href="https://github.com/zjuwhw"><span class="icon icon--github"><svg viewBox="0 0 16 16"><path fill="#828282" d="M7.999,0.431c-4.285,0-7.76,3.474-7.76,7.761 c0,3.428,2.223,6.337,5.307,7.363c0.388,0.071,0.53-0.168,0.53-0.374c0-0.184-0.007-0.672-0.01-1.32 c-2.159,0.469-2.614-1.04-2.614-1.04c-0.353-0.896-0.862-1.135-0.862-1.135c-0.705-0.481,0.053-0.472,0.053-0.472 c0.779,0.055,1.189,0.8,1.189,0.8c0.692,1.186,1.816,0.843,2.258,0.645c0.071-0.502,0.271-0.843,0.493-1.037 C4.86,11.425,3.049,10.76,3.049,7.786c0-0.847,0.302-1.54,0.799-2.082C3.768,5.507,3.501,4.718,3.924,3.65 c0,0,0.652-0.209,2.134,0.796C6.677,4.273,7.34,4.187,8,4.184c0.659,0.003,1.323,0.089,1.943,0.261 c1.482-1.004,2.132-0.796,2.132-0.796c0.423,1.068,0.157,1.857,0.077,2.054c0.497,0.542,0.798,1.235,0.798,2.082 c0,2.981-1.814,3.637-3.543,3.829c0.279,0.24,0.527,0.713,0.527,1.437c0,1.037-0.01,1.874-0.01,2.129 c0,0.208,0.14,0.449,0.534,0.373c3.081-1.028,5.302-3.935,5.302-7.362C15.76,3.906,12.285,0.431,7.999,0.431z"/></svg>
</span><span class="username">zjuwhw</span></a>

          </li>
          

          
          <li>
            <a href="https://twitter.com/huanwei_wang"><span class="icon icon--twitter"><svg viewBox="0 0 16 16"><path fill="#828282" d="M15.969,3.058c-0.586,0.26-1.217,0.436-1.878,0.515c0.675-0.405,1.194-1.045,1.438-1.809c-0.632,0.375-1.332,0.647-2.076,0.793c-0.596-0.636-1.446-1.033-2.387-1.033c-1.806,0-3.27,1.464-3.27,3.27 c0,0.256,0.029,0.506,0.085,0.745C5.163,5.404,2.753,4.102,1.14,2.124C0.859,2.607,0.698,3.168,0.698,3.767 c0,1.134,0.577,2.135,1.455,2.722C1.616,6.472,1.112,6.325,0.671,6.08c0,0.014,0,0.027,0,0.041c0,1.584,1.127,2.906,2.623,3.206 C3.02,9.402,2.731,9.442,2.433,9.442c-0.211,0-0.416-0.021-0.615-0.059c0.416,1.299,1.624,2.245,3.055,2.271 c-1.119,0.877-2.529,1.4-4.061,1.4c-0.264,0-0.524-0.015-0.78-0.046c1.447,0.928,3.166,1.469,5.013,1.469 c6.015,0,9.304-4.983,9.304-9.304c0-0.142-0.003-0.283-0.009-0.423C14.976,4.29,15.531,3.714,15.969,3.058z"/></svg>
</span><span class="username">huanwei_wang</span></a>

          </li>
          
        </ul>
      </div>

      <div class="footer-col footer-col-3">
        <p>This is Huanwei Wang's personal blog. Powered by Jekyll and Host in Github.
</p>
      </div>
    </div>

  </div>

</footer>


  </body>

</html>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    processEscapes: true
  }
});
</script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js?config=TeX-AMS-MML_HTMLorMML" type="text/javascript"></script>
