<!DOCTYPE html>
<html>

  <head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  <title>Genetic Relationship Matrix (GRM)</title>
  <meta name="description" content="GRM, short for genetic relationship matrix, is a important matrix used in genetic analysis of human complex traits.">

  <link rel="stylesheet" href="/css/main.css">
  <link rel="canonical" href="http://localhost:4000/2021/08/20/GRM.html">
  <link rel="alternate" type="application/rss+xml" title="Huanwei Wang's blog" href="http://localhost:4000/feed.xml">
</head>


  <body>

    <!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-121065275-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-121065275-1');
</script>



    <header class="site-header">

  <div class="wrapper">

    <a class="site-title" href="/">Huanwei Wang's blog</a>

    <nav class="site-nav">
      <a href="#" class="menu-icon">
        <svg viewBox="0 0 18 15">
          <path fill="#424242" d="M18,1.484c0,0.82-0.665,1.484-1.484,1.484H1.484C0.665,2.969,0,2.304,0,1.484l0,0C0,0.665,0.665,0,1.484,0 h15.031C17.335,0,18,0.665,18,1.484L18,1.484z"/>
          <path fill="#424242" d="M18,7.516C18,8.335,17.335,9,16.516,9H1.484C0.665,9,0,8.335,0,7.516l0,0c0-0.82,0.665-1.484,1.484-1.484 h15.031C17.335,6.031,18,6.696,18,7.516L18,7.516z"/>
          <path fill="#424242" d="M18,13.516C18,14.335,17.335,15,16.516,15H1.484C0.665,15,0,14.335,0,13.516l0,0 c0-0.82,0.665-1.484,1.484-1.484h15.031C17.335,12.031,18,12.696,18,13.516L18,13.516z"/>
        </svg>
      </a>

      <div class="trigger">
        
          
        
          
          <a class="page-link" href="/about/">About</a>
          
        
          
        
          
        
          
        
          
        
          
        
          
          <a class="page-link" href="/links/">Links</a>
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
          <a class="page-link" href="/tags/">Tags</a>
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
      </div>
    </nav>

  </div>

</header>


    <div class="page-content">
      <div class="wrapper">
        <article class="post" itemscope itemtype="http://schema.org/BlogPosting">

  <header class="post-header">
    <h1 class="post-title" itemprop="name headline">Genetic Relationship Matrix (GRM)</h1>
    <p class="post-meta"><time datetime="2021-08-20T00:00:00+10:00" itemprop="datePublished">Aug 20, 2021</time></p>
  </header>

  <div class="post-content" itemprop="articleBody">
    <p>GRM, short for genetic relationship matrix, is a important matrix used in genetic analysis of human complex traits.</p>

<p>This blog is my study note for GRM.</p>

<h2 id="mixed-linear-model-mlm">Mixed linear model (MLM)</h2>

<p>Based on <a href="https://doi.org/10.1016/j.ajhg.2010.11.011">Yang et al., AJHG, 2011</a></p>

\[\pmb{y}_{n \times 1}=\pmb{X}_{n \times k}\pmb{\beta}_{k \times 1}+\pmb{W}_{n \times m}\pmb{u}_{m \times 1}+\pmb{\epsilon}_{n \times 1}\]

<ul>
  <li>$\pmb{y}$ is a vector of phenotypes; n: the number of individuals;</li>
  <li>$\pmb{X}$ is a matrix for variables with fixed effects; k: the number of fixed effect factors;</li>
  <li>$\pmb{\beta}$ is a vector of fixed effects;</li>
  <li>$\pmb{u}$ is a vector of random effects; $\pmb{u} \sim N(0, \pmb{I}\sigma_u^2)$, where $\pmb{I}$ is a $n \times n$ identity matrix, and m i the number of SNPs;</li>
  <li>$\pmb{W}$ is a matrix for standardized genotype matrix, where its element for i-th SNP of j-th individual $w_{ij}=(x_{ij}-2p_i)/\sqrt{2p_i(1-p_i)}$ with $x_{ij}$ being the genotype value coded as the number of copies of the reference alleles {0, 1, 2};</li>
</ul>

\[Var(\pmb{y})=\pmb{V}_{n \times n}=\pmb{W}\pmb{W}^T\sigma_u^2+\pmb{I}\sigma_{\epsilon}^2\]

<p>If define $\sigma_g^2=m\sigma_u^2$ (the variance explained by all the SNPs), $\pmb{g}=\pmb{W}\pmb{u}$ and $\pmb{A}=\pmb{W}\pmb{W}^T/m$ (genetic relationship matrix (GRM) between individuals),</p>

\[\pmb{y}=\pmb{X}\pmb{\beta}+\pmb{g}+\pmb{\sigma}\text{ and }\pmb{V}=\pmb{A}\sigma_g^2+\pmb{I}\sigma_{\epsilon}^2\]

<h2 id="practice">Practice</h2>

<h3 id="simulate-the-genotype-matrix">Simulate the genotype matrix</h3>

<p>I use the R language to simulate a genotype matrix for <code class="language-plaintext highlighter-rouge">n</code> individuals and <code class="language-plaintext highlighter-rouge">m</code> SNPs</p>

<p><em>R code</em></p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>n=5; m=3
set.seed(10)
p = runif(m, min=0.2, max=0.5) ### allele frequency were draw from a uniform distribution
x_A1 = t(replicate(n, rbinom(m, 1, p)))   ### for the plink ped file
x_A2 = t(replicate(n, rbinom(m, 1, p)))
x = x_A1 + x_A2
colnames(x) = paste("SNP",1:m,sep="")
rownames(x) = paste("indi", 1:n, sep="")
x
#### now we save the matrix x, convert to plink bfile, and then calculate GRM using GCTA
x_A1 = ifelse(x_A1==0, "a", "A")
x_A2 = ifelse(x_A2==0, "a", "A")
dir.create("toy")
write.table(cbind(data.frame(FID=rownames(x), IID=rownames(x), 0, 0, 0, 0), cbind(x_A1, x_A2)[,order(c(1:m, 1:m+0.5))]), "toy/toy.ped", quote=F, sep="\t", row.names=F, col.names=F)
write.table(data.frame(CHR=1, SNP=colnames(x), cM=0, BP=10000+1:ncol(x)), "toy/toy.map", quote=F, sep="\t", row.names=F, col.names=F)
</code></pre></div></div>

<p><em>Output</em></p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>##       SNP1 SNP2 SNP3
## indi1    1    1    1
## indi2    0    1    0
## indi3    0    0    1
## indi4    0    1    0
## indi5    0    0    0
</code></pre></div></div>

<h3 id="after-standardization">After standardization</h3>

<p><em>R code</em></p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>p_hat = apply(x, 2, sum)/(2*n)
w = apply(rbind(x,p_hat), 2, function(x) (x-2*x[length(x)])/sqrt(2*x[length(x)]*(1-x[length(x)])))[1:n,]
w
</code></pre></div></div>

<p><em>Output</em></p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>##             SNP1       SNP2       SNP3
## indi1  1.8856181  0.6172134  1.0606602
## indi2 -0.4714045  0.6172134 -0.7071068
## indi3 -0.4714045 -0.9258201  1.0606602
## indi4 -0.4714045  0.6172134 -0.7071068
## indi5 -0.4714045 -0.9258201 -0.7071068
</code></pre></div></div>

<h3 id="calculate-the-grm">Calculate the GRM</h3>

<p>The diagonal elements in GRM</p>

<p><em>R code</em></p>

<pre><code class="language-{r}">A = w %*% t(w) /m
A
</code></pre>

<p><em>Output</em></p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>##            indi1       indi2      indi3       indi4       indi5
## indi1  1.6871693 -0.41931217 -0.1117725 -0.41931217 -0.73677249
## indi2 -0.4193122  0.36772487 -0.3664021  0.36772487  0.05026455
## indi3 -0.1117725 -0.36640212  0.7347884 -0.36640212  0.10978836
## indi4 -0.4193122  0.36772487 -0.3664021  0.36772487  0.05026455
## indi5 -0.7367725  0.05026455  0.1097884  0.05026455  0.52645503
</code></pre></div></div>

<p><em>R code</em></p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>diag(A)
</code></pre></div></div>

<p><em>Output</em></p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>##     indi1     indi2     indi3     indi4     indi5 
## 1.6871693 0.3677249 0.7347884 0.3677249 0.5264550
</code></pre></div></div>

<p>The off-diagonal elelments in GRM</p>

<p><em>R code</em></p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>A[row(A)&lt;col(A)]
</code></pre></div></div>

<p><em>Output</em></p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>##  [1] -0.41931217 -0.11177249 -0.36640212 -0.41931217  0.36772487 -0.36640212
##  [7] -0.73677249  0.05026455  0.10978836  0.05026455
</code></pre></div></div>

<p><em>R code</em></p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>A[row(A)&gt;col(A)]
</code></pre></div></div>

<p><em>Output</em></p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>##  [1] -0.41931217 -0.11177249 -0.41931217 -0.73677249 -0.36640212  0.36772487
##  [7]  0.05026455 -0.36640212  0.10978836  0.05026455
</code></pre></div></div>

<h3 id="run-the-make-grm-using-gcta">Run the “make-grm” using GCTA</h3>

<p>Save the genotype matrix in Plink bfile formate</p>

<p><em>bash code</em></p>

<pre><code class="language-{bash}">~/Software/PLINK2/plink --ped toy/toy.ped --map toy/toy.map --make-bed --out toy/toy   ### generate PLINK bfiles: .bim, .fam, .bed
</code></pre>

<p><em>log message</em></p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## PLINK v1.90b4.4 64-bit (21 May 2017)           www.cog-genomics.org/plink/1.9/
## (C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3
## Logging to toy/toy.log.
## Options in effect:
##   --make-bed
##   --map toy/toy.map
##   --out toy/toy
##   --ped toy/toy.ped
## 
## 16384 MB RAM detected; reserving 8192 MB for main workspace.
## Scanning .ped file... 0%20%40%60%80%100%
.ped scan complete (for binary autoconversion).
## Performing single-pass .bed write (3 variants, 5 people).
## 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%
--file: toy/toy-temporary.bed + toy/toy-temporary.bim + toy/toy-temporary.fam
## written.
## 3 variants loaded from .bim file.
## 5 people (0 males, 0 females, 5 ambiguous) loaded from .fam.
## Ambiguous sex IDs written to toy/toy.nosex .
## Using 1 thread (no multithreaded calculations invoked).
## Before main variant filters, 5 founders and 0 nonfounders present.
## Calculating allele frequencies... 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%95%96%97%98%99% done.
## 3 variants and 5 people pass filters and QC.
## Note: No phenotypes present.
## --make-bed to toy/toy.bed + toy/toy.bim + toy/toy.fam ... 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%95%96%97%98%99%done.
</code></pre></div></div>

<p><em>bash code</em></p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>~/Software/gcta_1.93.2beta_mac/gcta64 --bfile toy/toy --make-grm --out toy/toy.grm  ### generate grm using GCTA
</code></pre></div></div>

<p><em>log message</em></p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## *******************************************************************
## * Genome-wide Complex Trait Analysis (GCTA)
## * version 1.93.2 beta Mac
## * (C) 2010-present, Jian Yang, The University of Queensland
## * Please report bugs to Jian Yang &lt;jian.yang.qt@gmail.com&gt;
## *******************************************************************
## Analysis started at 16:41:12 AEST on Tue Aug 03 2021.
## Hostname: imb20-015312-lt
## 
## Options: 
##  
## --bfile toy/toy 
## --make-grm 
## --out toy/toy.grm 
## 
## Note: GRM is computed using the SNPs on the autosome.
## Reading PLINK FAM file from [toy/toy.fam]...
## 5 individuals to be included from FAM file.
## 5 individuals to be included. 0 males, 0 females, 5 unknown.
## Reading PLINK BIM file from [toy/toy.bim]...
## 3 SNPs to be included from BIM file(s).
## Computing the genetic relationship matrix (GRM) v2 ...
## Subset 1/1, no. subject 1-5
##   5 samples, 3 markers, 15 GRM elements
## IDs for the GRM file has been saved in the file [toy/toy.grm.grm.id]
## Computing GRM...
##   100% finished in 0.0 sec
## 3 SNPs have been processed.
##   Used 3 valid SNPs.
## The GRM computation is completed.
## Saving GRM...
## GRM has been saved in the file [toy/toy.grm.grm.bin]
## Number of SNPs in each pair of individuals has been saved in the file [toy/toy.grm.grm.N.bin]
## 
## Analysis finished at 16:41:12 AEST on Tue Aug 03 2021
## Overall computational time: 0.00 sec.
</code></pre></div></div>

<h3 id="read-gcta-grm-output-in-r">Read GCTA-GRM output in R</h3>

<p><em>R code</em></p>

<pre><code class="language-{r}">### ReadGRMBin function copied from GCTA website
ReadGRMBin=function(prefix, AllN=F, size=4){
  sum_i=function(i){
    return(sum(1:i))
  }
  BinFileName=paste(prefix,".grm.bin",sep="")
  NFileName=paste(prefix,".grm.N.bin",sep="")
  IDFileName=paste(prefix,".grm.id",sep="")
  id = read.table(IDFileName)
  n=dim(id)[1]
  BinFile=file(BinFileName, "rb");
  grm=readBin(BinFile, n=n*(n+1)/2, what=numeric(0), size=size)
  NFile=file(NFileName, "rb");
  if(AllN==T){
    N=readBin(NFile, n=n*(n+1)/2, what=numeric(0), size=size)
  }
  else N=readBin(NFile, n=1, what=numeric(0), size=size)
  i=sapply(1:n, sum_i)
  return(list(diag=grm[i], off=grm[-i], id=id, N=N))
}
data = ReadGRMBin("toy/toy.grm")
data
</code></pre>
<p><em>Output</em></p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## $diag
## [1] 1.6871693 0.3677249 0.7347884 0.3677249 0.5264550
## 
## $off
##  [1] -0.41931218 -0.11177249 -0.36640212 -0.41931218  0.36772487 -0.36640212
##  [7] -0.73677248  0.05026455  0.10978836  0.05026455
## 
## $id
##      V1    V2
## 1 indi1 indi1
## 2 indi2 indi2
## 3 indi3 indi3
## 4 indi4 indi4
## 5 indi5 indi5
## 
## $N
## [1] 3
</code></pre></div></div>

  </div>

</article>

      </div>
    </div>

    <footer class="site-footer">

  <div class="wrapper">


    <div class="footer-col-wrapper">
      <div class="footer-col footer-col-1">
        <ul class="contact-list">
          <li>Huanwei Wang's blog</li>
          <li><a href="mailto:zju_whw@163.com">zju_whw@163.com</a></li>
        </ul>
      </div>

      <div class="footer-col footer-col-2">
        <ul class="social-media-list">
          
          <li>
            <a href="https://github.com/zjuwhw"><span class="icon icon--github"><svg viewBox="0 0 16 16"><path fill="#828282" d="M7.999,0.431c-4.285,0-7.76,3.474-7.76,7.761 c0,3.428,2.223,6.337,5.307,7.363c0.388,0.071,0.53-0.168,0.53-0.374c0-0.184-0.007-0.672-0.01-1.32 c-2.159,0.469-2.614-1.04-2.614-1.04c-0.353-0.896-0.862-1.135-0.862-1.135c-0.705-0.481,0.053-0.472,0.053-0.472 c0.779,0.055,1.189,0.8,1.189,0.8c0.692,1.186,1.816,0.843,2.258,0.645c0.071-0.502,0.271-0.843,0.493-1.037 C4.86,11.425,3.049,10.76,3.049,7.786c0-0.847,0.302-1.54,0.799-2.082C3.768,5.507,3.501,4.718,3.924,3.65 c0,0,0.652-0.209,2.134,0.796C6.677,4.273,7.34,4.187,8,4.184c0.659,0.003,1.323,0.089,1.943,0.261 c1.482-1.004,2.132-0.796,2.132-0.796c0.423,1.068,0.157,1.857,0.077,2.054c0.497,0.542,0.798,1.235,0.798,2.082 c0,2.981-1.814,3.637-3.543,3.829c0.279,0.24,0.527,0.713,0.527,1.437c0,1.037-0.01,1.874-0.01,2.129 c0,0.208,0.14,0.449,0.534,0.373c3.081-1.028,5.302-3.935,5.302-7.362C15.76,3.906,12.285,0.431,7.999,0.431z"/></svg>
</span><span class="username">zjuwhw</span></a>

          </li>
          

          
          <li>
            <a href="https://twitter.com/huanwei_wang"><span class="icon icon--twitter"><svg viewBox="0 0 16 16"><path fill="#828282" d="M15.969,3.058c-0.586,0.26-1.217,0.436-1.878,0.515c0.675-0.405,1.194-1.045,1.438-1.809c-0.632,0.375-1.332,0.647-2.076,0.793c-0.596-0.636-1.446-1.033-2.387-1.033c-1.806,0-3.27,1.464-3.27,3.27 c0,0.256,0.029,0.506,0.085,0.745C5.163,5.404,2.753,4.102,1.14,2.124C0.859,2.607,0.698,3.168,0.698,3.767 c0,1.134,0.577,2.135,1.455,2.722C1.616,6.472,1.112,6.325,0.671,6.08c0,0.014,0,0.027,0,0.041c0,1.584,1.127,2.906,2.623,3.206 C3.02,9.402,2.731,9.442,2.433,9.442c-0.211,0-0.416-0.021-0.615-0.059c0.416,1.299,1.624,2.245,3.055,2.271 c-1.119,0.877-2.529,1.4-4.061,1.4c-0.264,0-0.524-0.015-0.78-0.046c1.447,0.928,3.166,1.469,5.013,1.469 c6.015,0,9.304-4.983,9.304-9.304c0-0.142-0.003-0.283-0.009-0.423C14.976,4.29,15.531,3.714,15.969,3.058z"/></svg>
</span><span class="username">huanwei_wang</span></a>

          </li>
          
        </ul>
      </div>

      <div class="footer-col footer-col-3">
        <p>This is Huanwei Wang's personal blog. Powered by Jekyll and Host in Github.
</p>
      </div>
    </div>

  </div>

</footer>


  </body>

</html>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    processEscapes: true
  }
});
</script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js?config=TeX-AMS-MML_HTMLorMML" type="text/javascript"></script>
