<!DOCTYPE html>
<html>

  <head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  <title>Recombination and Recombination Rate</title>
  <meta name="description" content="Recently, I read the Adam Auton’s PhD Thesis “The Estimation of Recombination Rates from Population Genetic Data”, this blog is its notes.">

  <link rel="stylesheet" href="/css/main.css">
  <link rel="canonical" href="http://localhost:4000/2017/09/21/recombination.html">
  <link rel="alternate" type="application/rss+xml" title="Huanwei Wang's blog" href="http://localhost:4000/feed.xml">
</head>


  <body>

    <!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-121065275-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-121065275-1');
</script>



    <header class="site-header">

  <div class="wrapper">

    <a class="site-title" href="/">Huanwei Wang's blog</a>

    <nav class="site-nav">
      <a href="#" class="menu-icon">
        <svg viewBox="0 0 18 15">
          <path fill="#424242" d="M18,1.484c0,0.82-0.665,1.484-1.484,1.484H1.484C0.665,2.969,0,2.304,0,1.484l0,0C0,0.665,0.665,0,1.484,0 h15.031C17.335,0,18,0.665,18,1.484L18,1.484z"/>
          <path fill="#424242" d="M18,7.516C18,8.335,17.335,9,16.516,9H1.484C0.665,9,0,8.335,0,7.516l0,0c0-0.82,0.665-1.484,1.484-1.484 h15.031C17.335,6.031,18,6.696,18,7.516L18,7.516z"/>
          <path fill="#424242" d="M18,13.516C18,14.335,17.335,15,16.516,15H1.484C0.665,15,0,14.335,0,13.516l0,0 c0-0.82,0.665-1.484,1.484-1.484h15.031C17.335,12.031,18,12.696,18,13.516L18,13.516z"/>
        </svg>
      </a>

      <div class="trigger">
        
          
        
          
          <a class="page-link" href="/about/">About</a>
          
        
          
        
          
        
          
        
          
        
          
        
          
          <a class="page-link" href="/links/">Links</a>
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
          <a class="page-link" href="/tags/">Tags</a>
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
      </div>
    </nav>

  </div>

</header>


    <div class="page-content">
      <div class="wrapper">
        <article class="post" itemscope itemtype="http://schema.org/BlogPosting">

  <header class="post-header">
    <h1 class="post-title" itemprop="name headline">Recombination and Recombination Rate</h1>
    <p class="post-meta"><time datetime="2017-09-21T00:00:00+10:00" itemprop="datePublished">Sep 21, 2017</time></p>
  </header>

  <div class="post-content" itemprop="articleBody">
    <p>Recently, I read the <a href="http://adamauton.com/">Adam Auton</a>’s PhD Thesis <a href="http://adamauton.com/adam_auton_thesis.pdf">“The Estimation of Recombination Rates from Population Genetic Data”</a>, this blog is its notes.</p>

<h2 id="recombination">Recombination</h2>

<p><a href="https://en.wikipedia.org/wiki/Genetic_recombination">Homologous recombination</a> is the process by which a pair of homologous DNA sequences exchange some portion of their DNA. In eukaryotes, genetic recombination during meiosis can lead to a novel set of genetic information that can be passed on from the parents to the offspring</p>

<p>The recombination was found by <a href="https://en.wikipedia.org/wiki/Thomas_Hunt_Morgan">Thomas Hunt Morgan</a> in 1911, when studying the chromosome theory of heredity. And his student <a href="https://en.wikipedia.org/wiki/Alfred_Sturtevant">Sturtevant</a> uses crossover data to construct the first genetic map in drosophila. Read more <a href="https://www.nature.com/scitable/topicpage/thomas-hunt-morgan-genetic-recombination-and-gene-496">here</a>.</p>

<p>A model for the process by which recombination occurs in eukaryotes was proposed in 1964 by Robin Holliday (HOLLIDAY 1964). It shows two possible resolutions, known as Gene Conversion and Crossover. When talking about recombination, people always mean crossover as it leaves a much larger singal easy to trace.</p>

<p><img src="/images/recombination_holliday.png" alt="" /></p>

<h2 id="recombination-rate-centimorgan-cm">Recombination Rate (centimorgan, cM)</h2>

<p>A commonly used unit of recombination rate is the <a href="https://en.wikipedia.org/wiki/Centimorgan">centimorgan</a>, which is defined as a 1% chance that two loci will be separated by a recombination event in one generation.</p>

<p>In eukaryotes, there is a notable uniformity in the total recombination map length regardless of genome size (AWADALLA 2003). The average recombination rate is therefore negatively correlated with genome size.</p>

<p><img src="/images/recombination.png" alt="" /></p>

<h2 id="recombination-rate-estimation">Recombination Rate Estimation</h2>

<p>There are some methods to estimate the reombination rate.</p>

<h3 id="genotyping-a-large-number-of-individuals-in-families">Genotyping a large number of individuals in families</h3>

<p>In 2002, deCODE company (KONG et al. 2002) constructed a map from a large pedigree study of 869 individuals in 146 Icelandic families. They found</p>

<ul>
  <li>The shorter chromosomes tended to have higher recombination rates than the larger chromosomes</li>
  <li>There was also a high degree of rate heterogeneity within chromosomes</li>
  <li>an average of 1.65 times longer in females than in males</li>
</ul>

<h3 id="sperm-typing">Sperm Typing</h3>

<p>Sperm typing is to search for recombination sequences in the sperm product of male individuals.</p>

<h3 id="mhc-region">MHC Region</h3>

<p>A 216kb region of the major histocompatibility complex (MHC) on chromosome 6 has been extensively studied by sperm typing (JEFFREYS et al. 2001)</p>

<p>The rate of recombination in these regions could be hundred or thousands of times that of the surrounding regions, and hence these areas were dubbed <strong>recombination hotspots</strong>.</p>

<h3 id="population-data">Population data</h3>

<p>Recombination information can also be seem in the population genetic data.</p>

<p>Recombination events break down the amout of LD between loci. However, it is unclear how such statistics can be directly related to the underlying recombination rate (DEVLIN and RISCH 1995).</p>

<p>One early and simple method is known as <strong>four-gamete test</strong> (WEIR 1979). But it has very low power.</p>

<p>Nowadays, people, such as 23andme, use population data, such as HapMap to estimate the recombination rate using more complicated model and algorithms.</p>

<h3 id="the-genetic-map-provide-by-impute">The genetic map provide by IMPUTE</h3>

<p>The imputation software <a href="https://mathgen.stats.ox.ac.uk/impute/impute_v2.html">IMPUTE2</a> provide the genetic map for human in its <a href="https://mathgen.stats.ox.ac.uk/impute/1000GP_Phase3.html">website</a>.</p>

<p>We can have use its data to a look the recombination across the human genome.</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>library(data.table)
data = fread("genetic_map_chr1_combined_b37.txt")
plot(data$"position",data$"COMBINED_rate(cM/Mb)",type="l", xlab="chr1 (bp)", ylab="Recombination Rate(cM/Mb)")
</code></pre></div></div>

<p><img src="/images/recombination_chr1.png" alt="" /></p>

  </div>

</article>

      </div>
    </div>

    <footer class="site-footer">

  <div class="wrapper">


    <div class="footer-col-wrapper">
      <div class="footer-col footer-col-1">
        <ul class="contact-list">
          <li>Huanwei Wang's blog</li>
          <li><a href="mailto:zju_whw@163.com">zju_whw@163.com</a></li>
        </ul>
      </div>

      <div class="footer-col footer-col-2">
        <ul class="social-media-list">
          
          <li>
            <a href="https://github.com/zjuwhw"><span class="icon icon--github"><svg viewBox="0 0 16 16"><path fill="#828282" d="M7.999,0.431c-4.285,0-7.76,3.474-7.76,7.761 c0,3.428,2.223,6.337,5.307,7.363c0.388,0.071,0.53-0.168,0.53-0.374c0-0.184-0.007-0.672-0.01-1.32 c-2.159,0.469-2.614-1.04-2.614-1.04c-0.353-0.896-0.862-1.135-0.862-1.135c-0.705-0.481,0.053-0.472,0.053-0.472 c0.779,0.055,1.189,0.8,1.189,0.8c0.692,1.186,1.816,0.843,2.258,0.645c0.071-0.502,0.271-0.843,0.493-1.037 C4.86,11.425,3.049,10.76,3.049,7.786c0-0.847,0.302-1.54,0.799-2.082C3.768,5.507,3.501,4.718,3.924,3.65 c0,0,0.652-0.209,2.134,0.796C6.677,4.273,7.34,4.187,8,4.184c0.659,0.003,1.323,0.089,1.943,0.261 c1.482-1.004,2.132-0.796,2.132-0.796c0.423,1.068,0.157,1.857,0.077,2.054c0.497,0.542,0.798,1.235,0.798,2.082 c0,2.981-1.814,3.637-3.543,3.829c0.279,0.24,0.527,0.713,0.527,1.437c0,1.037-0.01,1.874-0.01,2.129 c0,0.208,0.14,0.449,0.534,0.373c3.081-1.028,5.302-3.935,5.302-7.362C15.76,3.906,12.285,0.431,7.999,0.431z"/></svg>
</span><span class="username">zjuwhw</span></a>

          </li>
          

          
          <li>
            <a href="https://twitter.com/huanwei_wang"><span class="icon icon--twitter"><svg viewBox="0 0 16 16"><path fill="#828282" d="M15.969,3.058c-0.586,0.26-1.217,0.436-1.878,0.515c0.675-0.405,1.194-1.045,1.438-1.809c-0.632,0.375-1.332,0.647-2.076,0.793c-0.596-0.636-1.446-1.033-2.387-1.033c-1.806,0-3.27,1.464-3.27,3.27 c0,0.256,0.029,0.506,0.085,0.745C5.163,5.404,2.753,4.102,1.14,2.124C0.859,2.607,0.698,3.168,0.698,3.767 c0,1.134,0.577,2.135,1.455,2.722C1.616,6.472,1.112,6.325,0.671,6.08c0,0.014,0,0.027,0,0.041c0,1.584,1.127,2.906,2.623,3.206 C3.02,9.402,2.731,9.442,2.433,9.442c-0.211,0-0.416-0.021-0.615-0.059c0.416,1.299,1.624,2.245,3.055,2.271 c-1.119,0.877-2.529,1.4-4.061,1.4c-0.264,0-0.524-0.015-0.78-0.046c1.447,0.928,3.166,1.469,5.013,1.469 c6.015,0,9.304-4.983,9.304-9.304c0-0.142-0.003-0.283-0.009-0.423C14.976,4.29,15.531,3.714,15.969,3.058z"/></svg>
</span><span class="username">huanwei_wang</span></a>

          </li>
          
        </ul>
      </div>

      <div class="footer-col footer-col-3">
        <p>This is Huanwei Wang's personal blog. Powered by Jekyll and Host in Github.
</p>
      </div>
    </div>

  </div>

</footer>


  </body>

</html>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    processEscapes: true
  }
});
</script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js?config=TeX-AMS-MML_HTMLorMML" type="text/javascript"></script>
