<!DOCTYPE html>
<html>

  <head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  <title>Genomic Alteration名词解释</title>
  <meta name="description" content="Glossary of Genomic Alteration">

  <link rel="stylesheet" href="/css/main.css">
  <link rel="canonical" href="http://localhost:4000/2015/10/15/genome_alteration.html">
  <link rel="alternate" type="application/rss+xml" title="Huanwei Wang's blog" href="http://localhost:4000/feed.xml">
</head>


  <body>

    <!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-121065275-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-121065275-1');
</script>



    <header class="site-header">

  <div class="wrapper">

    <a class="site-title" href="/">Huanwei Wang's blog</a>

    <nav class="site-nav">
      <a href="#" class="menu-icon">
        <svg viewBox="0 0 18 15">
          <path fill="#424242" d="M18,1.484c0,0.82-0.665,1.484-1.484,1.484H1.484C0.665,2.969,0,2.304,0,1.484l0,0C0,0.665,0.665,0,1.484,0 h15.031C17.335,0,18,0.665,18,1.484L18,1.484z"/>
          <path fill="#424242" d="M18,7.516C18,8.335,17.335,9,16.516,9H1.484C0.665,9,0,8.335,0,7.516l0,0c0-0.82,0.665-1.484,1.484-1.484 h15.031C17.335,6.031,18,6.696,18,7.516L18,7.516z"/>
          <path fill="#424242" d="M18,13.516C18,14.335,17.335,15,16.516,15H1.484C0.665,15,0,14.335,0,13.516l0,0 c0-0.82,0.665-1.484,1.484-1.484h15.031C17.335,12.031,18,12.696,18,13.516L18,13.516z"/>
        </svg>
      </a>

      <div class="trigger">
        
          
        
          
          <a class="page-link" href="/about/">About</a>
          
        
          
        
          
        
          
        
          
        
          
        
          
          <a class="page-link" href="/links/">Links</a>
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
          <a class="page-link" href="/tags/">Tags</a>
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
      </div>
    </nav>

  </div>

</header>


    <div class="page-content">
      <div class="wrapper">
        <article class="post" itemscope itemtype="http://schema.org/BlogPosting">

  <header class="post-header">
    <h1 class="post-title" itemprop="name headline">Genomic Alteration名词解释</h1>
    <p class="post-meta"><time datetime="2015-10-15T00:00:00+10:00" itemprop="datePublished">Oct 15, 2015</time></p>
  </header>

  <div class="post-content" itemprop="articleBody">
    <h1 id="glossary-of-genomic-alteration">Glossary of Genomic Alteration</h1>

<p><strong>1. <a href="http://www.cancer.gov/publications/dictionaries/cancer-terms?CdrID=446539">SNP (Single Nucleotide Polymorphism)</a>:</strong> The most common type of change in DNA (molecules inside cells that carry genetic information). SNPs occur when a single nucleotide (building block of DNA) is replaced with another. These changes may cause disease, and may affect how a person reacts to bacteria, viruses, drugs, and other substances. Also called single nucleotide polymorphism.</p>

<p><strong>2. <a href="http://www.genome.gov/glossary/index.cfm?id=99">Haplotype</a></strong>: A haplotype is a set of DNA variations, or polymorphisms, that tend to be inherited together. A haplotype can refer to a combination of alleles or to a set of single nucleotide polymorphisms (SNPs) found on the same chromosome. Information about haplotypes is being collected by the International HapMap Project and is used to investigate the influence of genes on disease.</p>

<p><strong>3. SNV:</strong> Single Nucleotide Variation.</p>

<p><strong>4. MNV:</strong> Multinucleotide variation.</p>

<p><strong>5. <a href="http://www.cancer.gov/publications/dictionaries/cancer-terms?CdrID=446539">Mutation</a></strong>: Any change in the DNA sequence of a cell. Mutations may be caused by mistakes during cell division, or they may be caused by exposure to DNA-damaging agents in the environment. Mutations can be harmful, beneficial, or have no effect. If they occur in cells that make eggs or sperm, they can be inherited; if mutations occur in other types of cells, they are not inherited. Certain mutations may lead to cancer or other diseases.</p>

<p><strong>6. <a href="http://www.genome.gov/glossary/index.cfm?id=190">Substitution</a></strong>: Substitution is a type of mutation where one base pair is replaced by a different base pair. The term also refers to the replacement of one amino acid in a protein with a different amino acid.</p>

<p><strong>7. <a href="http://www.genome.gov/glossary/index.cfm?id=156">Point mutation</a></strong>: A point mutation is when a single base pair is altered. Point mutations can have one of three effects. First, the base substitution can be a silent mutation where the altered codon corresponds to the same amino acid. Second, the base substitution can be a missense mutation where the altered codon corresponds to a different amino acid. Or third, the base substitution can be a nonsense mutation where the altered codon corresponds to a stop signal.</p>

<p><strong>8. <a href="http://ghr.nlm.nih.gov/glossary=silentmutation">Silent mutation</a>:</strong> A silent mutation alters DNA sequence, but has no apparent detectable effect on a phenotype or a function.</p>

<p><strong>9. <a href="http://www.genome.gov/glossary/index.cfm?id=127">Missense (Conservative/Non-conservative) mutation</a>:</strong> A missense mutation is when the change of a single base pair causes the substitution of a different amino acid in the resulting protein. This amino acid substitution may have no effect, or it may render the protein nonfunctional.</p>

<p><strong>10. <a href="http://www.genome.gov/glossary/index.cfm?id=138">Nonsense mutation</a></strong>: A nonsense mutation is the substitution of a single base pair that leads to the appearance of a stop codon where previously there was a codon specifying an amino acid. The presence of this premature stop codon results in the production of a shortened, and likely nonfunctional, protein.</p>

<p><strong>11. <a href="http://www.genome.gov/glossary/index.cfm?id=112">Insertion</a></strong>: Insertion is a type of mutation involving the addition of genetic material. An insertion mutation can be small, involving a single extra DNA base pair, or large, involving a piece of a chromosome.</p>

<p><strong>12. <a href="http://www.genome.gov/glossary/index.cfm?id=45">Deletion</a></strong>: Deletion is a type of mutation involving the loss of genetic material. It can be small, involving a single missing DNA base pair, or large, involving a piece of a chromosome.</p>

<p><strong>13. Indel:</strong> insertion and deletion.</p>

<p><strong>14. <a href="http://www.genome.gov/glossary/index.cfm?id=68">Frameshift mutation</a></strong>: A frameshift mutation is a type of mutation involving the insertion or deletion of a nucleotide in which the number of deleted base pairs is not divisible by three. “Divisible by three” is important because the cell reads a gene in groups of three bases. Each group of three bases corresponds to one of 20 different amino acids used to build a protein. If a mutation disrupts this reading frame, then the entire DNA sequence following the mutation will be read incorrectly.</p>

<p><strong>15. <a href="http://www.ncbi.nlm.nih.gov/pubmed/19360079">Somatic/Germline mutation</a>:</strong> Like all the cells that constitute the human body, a cancer cell is a direct descendant, through a lineage of mitotic cell divisions, of the fertilized egg from which the cancer patient developed and therefore carries a copy of its diploid genome. However, the DNA sequence of a cancer cell genome, and indeed of most normal cell genomes, has acquired a set of differences from its progenitor fertilized egg. These are collectively termed <strong>somatic mutations</strong> to distinguish them from <strong>germline mutations</strong> that are inherited from parents and transmitted to offspring,</p>

<p><img src="/images/Blog6_genome.alteration.glossary_mutations.png" alt="mutation" /></p>

<p><strong>16. <a href="http://www.ncbi.nlm.nih.gov/pubmed/19360079">Driver/Passager mutation</a>:</strong></p>

<p>All cancers arise as a result of somatically acquired changes in the DNA cancer cells. That does not mean, however, that all the somatic abnormalities present in a cancer genome have been involved in development of the cancer. Indeed, it is likely that some have made no contribution at all. To embody this concept, the terms “driver” and “passenger” mutation have been coined.</p>

<p>A <strong>driver mutation</strong> is causally implicated in oncognesis. It has conferred growth advantage on the cancer cell and has been positively selected in the microenvironment of the tissue in which the cancer arises. A driver mutation need not be required for maintenance of the final cancer (although it often is) but it must have been selected at some point along the lieage of cancer development shown in above figure.</p>

<p>A <strong>passenger mutation</strong> has not been selected, has not conferred clonal growth advantage and has therefore not contributed to cancer development. Passenger mutations are found within cancer genomes because somatic mutations without functional consequences often occur during cell divison. Thus, a cell that acquires a driver mutation will already have biologically insert somatic mutations within its genome. These will be carried along in the clonal expansion that follows and therefore will be present in all cells of the final cancer.</p>

<p><strong>17. <a href="http://www.ncbi.nlm.nih.gov/dbvar/content/overview/">Structural variation (SV)</a>:</strong> Structural variation (SV) is generally defined as a region of DNA approximately 1 kb and larger in size and can include inversions and balanced translocations or genomic imbalances (insertions and deletions), commonly referred to as copy number variants (CNVs). These CNVs often overlap with segmental duplications, regions of DNA &gt;1 kb present more than once in the genome, copies of which are &gt;90% identical . If present at &gt;1% in a population a CNV may be referred to as copy number polymorphism (CNP).</p>

<p><strong>18. <a href="https://en.wikipedia.org/wiki/Chromosomal_rearrangement">Chromosomal rearrangement</a>:</strong> In genetics, a chromosomal rearrangement is a type of chromosome abnormality involving a change in the structure of the native chromosome. Such changes may involve several different classes of events, like deletions, duplications, inversions, and translocations.</p>

<p><strong>19. <a href="http://ghr.nlm.nih.gov/glossary=inversion">Inversion</a>:</strong> A chromosomal rearrangement in which a segment of genetic material is broken away from the chromosome, inverted from end to end, and re-inserted into the chromosome at the same breakage site. Balanced inversions (no net loss or gain of genetic material) are usually not associated with phenotypic abnormalities, although in some cases gene disruptions at the breakpoints can cause adverse phenotypic effects, including some known genetic diseases; unbalanced inversions (loss or gain of chromosome material) nearly always yield an abnormal phenotype.</p>

<p><strong>20. <a href="http://www.genome.gov/glossary/index.cfm?id=201">Translocation</a>:</strong> Translocation is a type of chromosomal abnormality in which a chromosome breaks and a portion of it reattaches to a different chromosome. Chromosomal translocations can be detected by analyzing karyotypes of the affected cells.</p>

<p><strong>21. <a href="http://www.genome.gov/glossary/index.cfm?id=40">Copy number variation (CNV)</a></strong>: A copy number variation (CNV) is when the number of copies of a particular gene varies from one individual to the next. Following the completion of the Human Genome Project, it became apparent that the genome experiences gains and losses of genetic material. The extent to which copy number variation contributes to human disease is not yet known. It has long been recognized that some cancers are associated with elevated copy numbers of particular genes.</p>

<p><strong>22. <a href="http://www.genome.gov/glossary/index.cfm?id=55">Duplication</a>:</strong> Duplication is a type of mutation that involves the production of one or more copies of a gene or region of a chromosome. Gene and chromosome duplications occur in all organisms, though they are especially prominent among plants. Gene duplication is an important mechanism by which evolution occurs.</p>

<p><strong>23. <a href="http://www.genome.gov/glossary/index.cfm?id=71">Gene amplification</a></strong>: Gene amplification is an increase in the number of copies of a gene sequence. Cancer cells sometimes produce multiple copies of genes in response to signals from other cells or their environment. The term also can refer to polymerase chain reaction (PCR), a laboratory technique that is used by scientists to amplify gene sequences in a test tube.</p>

<p><strong>24. <a href="http://www.genome.gov/glossary/index.cfm?id=89">Genetic variation</a>:</strong> Genetic variation refers to diversity in gene frequencies. Genetic variation can refer to differences between individuals or to differences between populations. Mutation is the ultimate source of genetic variation, but mechanisms such as sexual reproduction and genetic drift contribute to it as well.</p>

<p><strong>25.<a href="http://www.genome.gov/glossary/index.cfm?id=193">Tandem Repeat</a>:</strong> A tandem repeat is a sequence of two or more DNA base pairs that is repeated in such a way that the repeats lie adjacent to each other on the chromosome. Tandem repeats are generally associated with non-coding DNA. In some instances, the number of times the DNA sequence is repeated is variable. Such variable tandem repeats are used in DNA fingerprinting procedures.</p>

<h3 id="reference">Reference:</h3>

<ul>
  <li>
    <p><a href="http://www.cancer.gov/publications/dictionaries/cancer-terms">http://www.cancer.gov/publications/dictionaries/cancer-terms</a></p>
  </li>
  <li>
    <p><a href="http://www.genome.gov/glossary/">http://www.genome.gov/glossary/</a></p>
  </li>
  <li>
    <p><a href="https://en.wikipedia.org">https://en.wikipedia.org</a></p>
  </li>
  <li>
    <p><a href="http://ghr.nlm.nih.gov/">http://ghr.nlm.nih.gov/</a></p>
  </li>
  <li>
    <p><a href="http://www.ncbi.nlm.nih.gov/dbvar/content/overview/">http://www.ncbi.nlm.nih.gov/dbvar/content/overview/</a></p>
  </li>
  <li>
    <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/19360079">Michael R. Stratton et al., Nature 2009.</a></p>
  </li>
</ul>

  </div>

</article>

      </div>
    </div>

    <footer class="site-footer">

  <div class="wrapper">


    <div class="footer-col-wrapper">
      <div class="footer-col footer-col-1">
        <ul class="contact-list">
          <li>Huanwei Wang's blog</li>
          <li><a href="mailto:zju_whw@163.com">zju_whw@163.com</a></li>
        </ul>
      </div>

      <div class="footer-col footer-col-2">
        <ul class="social-media-list">
          
          <li>
            <a href="https://github.com/zjuwhw"><span class="icon icon--github"><svg viewBox="0 0 16 16"><path fill="#828282" d="M7.999,0.431c-4.285,0-7.76,3.474-7.76,7.761 c0,3.428,2.223,6.337,5.307,7.363c0.388,0.071,0.53-0.168,0.53-0.374c0-0.184-0.007-0.672-0.01-1.32 c-2.159,0.469-2.614-1.04-2.614-1.04c-0.353-0.896-0.862-1.135-0.862-1.135c-0.705-0.481,0.053-0.472,0.053-0.472 c0.779,0.055,1.189,0.8,1.189,0.8c0.692,1.186,1.816,0.843,2.258,0.645c0.071-0.502,0.271-0.843,0.493-1.037 C4.86,11.425,3.049,10.76,3.049,7.786c0-0.847,0.302-1.54,0.799-2.082C3.768,5.507,3.501,4.718,3.924,3.65 c0,0,0.652-0.209,2.134,0.796C6.677,4.273,7.34,4.187,8,4.184c0.659,0.003,1.323,0.089,1.943,0.261 c1.482-1.004,2.132-0.796,2.132-0.796c0.423,1.068,0.157,1.857,0.077,2.054c0.497,0.542,0.798,1.235,0.798,2.082 c0,2.981-1.814,3.637-3.543,3.829c0.279,0.24,0.527,0.713,0.527,1.437c0,1.037-0.01,1.874-0.01,2.129 c0,0.208,0.14,0.449,0.534,0.373c3.081-1.028,5.302-3.935,5.302-7.362C15.76,3.906,12.285,0.431,7.999,0.431z"/></svg>
</span><span class="username">zjuwhw</span></a>

          </li>
          

          
          <li>
            <a href="https://twitter.com/huanwei_wang"><span class="icon icon--twitter"><svg viewBox="0 0 16 16"><path fill="#828282" d="M15.969,3.058c-0.586,0.26-1.217,0.436-1.878,0.515c0.675-0.405,1.194-1.045,1.438-1.809c-0.632,0.375-1.332,0.647-2.076,0.793c-0.596-0.636-1.446-1.033-2.387-1.033c-1.806,0-3.27,1.464-3.27,3.27 c0,0.256,0.029,0.506,0.085,0.745C5.163,5.404,2.753,4.102,1.14,2.124C0.859,2.607,0.698,3.168,0.698,3.767 c0,1.134,0.577,2.135,1.455,2.722C1.616,6.472,1.112,6.325,0.671,6.08c0,0.014,0,0.027,0,0.041c0,1.584,1.127,2.906,2.623,3.206 C3.02,9.402,2.731,9.442,2.433,9.442c-0.211,0-0.416-0.021-0.615-0.059c0.416,1.299,1.624,2.245,3.055,2.271 c-1.119,0.877-2.529,1.4-4.061,1.4c-0.264,0-0.524-0.015-0.78-0.046c1.447,0.928,3.166,1.469,5.013,1.469 c6.015,0,9.304-4.983,9.304-9.304c0-0.142-0.003-0.283-0.009-0.423C14.976,4.29,15.531,3.714,15.969,3.058z"/></svg>
</span><span class="username">huanwei_wang</span></a>

          </li>
          
        </ul>
      </div>

      <div class="footer-col footer-col-3">
        <p>This is Huanwei Wang's personal blog. Powered by Jekyll and Host in Github.
</p>
      </div>
    </div>

  </div>

</footer>


  </body>

</html>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    processEscapes: true
  }
});
</script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js?config=TeX-AMS-MML_HTMLorMML" type="text/javascript"></script>
