<!DOCTYPE html>
<html>

  <head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  <title>Fastq格式</title>
  <meta name="description" content="Fastq是二代测序中常用的原始序列文件格式。">

  <link rel="stylesheet" href="/css/main.css">
  <link rel="canonical" href="http://localhost:4000/2016/08/13/fastq_format.html">
  <link rel="alternate" type="application/rss+xml" title="zjuwhw's blog" href="http://localhost:4000/feed.xml">
</head>


  <body>

    <header class="site-header">

  <div class="wrapper">

    <a class="site-title" href="/">zjuwhw's blog</a>

    <nav class="site-nav">
      <a href="#" class="menu-icon">
        <svg viewBox="0 0 18 15">
          <path fill="#424242" d="M18,1.484c0,0.82-0.665,1.484-1.484,1.484H1.484C0.665,2.969,0,2.304,0,1.484l0,0C0,0.665,0.665,0,1.484,0 h15.031C17.335,0,18,0.665,18,1.484L18,1.484z"/>
          <path fill="#424242" d="M18,7.516C18,8.335,17.335,9,16.516,9H1.484C0.665,9,0,8.335,0,7.516l0,0c0-0.82,0.665-1.484,1.484-1.484 h15.031C17.335,6.031,18,6.696,18,7.516L18,7.516z"/>
          <path fill="#424242" d="M18,13.516C18,14.335,17.335,15,16.516,15H1.484C0.665,15,0,14.335,0,13.516l0,0 c0-0.82,0.665-1.484,1.484-1.484h15.031C17.335,12.031,18,12.696,18,13.516L18,13.516z"/>
        </svg>
      </a>

      <div class="trigger">
        
          
          <a class="page-link" href="/about/">About</a>
          
        
          
        
          
        
          
          <a class="page-link" href="/links/">Links</a>
          
        
          
        
          
        
          
        
          
          <a class="page-link" href="/tags/">Tags</a>
          
        
          
        
          
        
          
        
      </div>
    </nav>

  </div>

</header>


    <div class="page-content">
      <div class="wrapper">
        <article class="post" itemscope itemtype="http://schema.org/BlogPosting">

  <header class="post-header">
    <h1 class="post-title" itemprop="name headline">Fastq格式</h1>
    <p class="post-meta"><time datetime="2016-08-13T00:00:00+10:00" itemprop="datePublished">Aug 13, 2016</time></p>
  </header>

  <div class="post-content" itemprop="articleBody">
    <p><a href="https://en.wikipedia.org/wiki/FASTQ_format">Fastq</a>是二代测序中常用的原始序列文件格式。</p>

<h2 id="fastq文件如何得到">Fastq文件如何得到？</h2>

<blockquote>
  <p>After the sequencing platform generates the sequencing images, the data are analyzed in five steps: image analysis, base calling, bcl conversion, sequence alignment, and variant analysis and counting. CASAVA performs the bcl conversion, sequence alignment, and variant analysis and counting steps, demultiplexes multiplexed samples during the bcl conversion step.
Bcl conversion — Converts *.bcl files into *.fastq.gz files (compressed FASTQ files) in CASAVA. Multiplexed samples are demultiplexed during this step.</p>
</blockquote>

<p>在测序平台产生测序图像之后，数据分析有5个步骤：图像分析，base calling，bcl文件转换，序列比对和变异分析。illumina的CASAVA软件进行bcl转换，序列比对和变异分析这些步骤。在bcl转换步骤中，多样本数据将会拆分。</p>

<p>Bcl转换 - 用CASAVA软件转换bcl文件为fastq.gz文件（压缩的fastq文件）。多样本数据将会拆分。</p>

<h2 id="示例">示例：</h2>

<div class="highlighter-rouge"><pre class="highlight"><code>@EAS139:136:FC706VJ:2:5:1000:12850 1:Y:18:ATCACG
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+
BBBBCCCC?&lt;A?BC?7@@???????DBBA@@@@A@@
</code></pre>
</div>

<ul>
  <li>第1行：header</li>
  <li>第2行：序列</li>
  <li>第3行：固定为“+”</li>
  <li>第4行：质量值（quality score）</li>
</ul>

<h2 id="header">Header</h2>

<p><code class="highlighter-rouge">@&lt;instrument&gt;:&lt;run number&gt;:&lt;flowcell ID&gt;:&lt;lane&gt;:&lt;tile&gt;:&lt;x-pos&gt;:&lt;y-pos&gt; &lt;read&gt;:&lt;is filtered&gt;:&lt;control number&gt;:&lt;index sequence&gt;</code></p>

<table>
  <thead>
    <tr>
      <th style="text-align: left">Element</th>
      <th style="text-align: left">Requirements</th>
      <th style="text-align: left">Description</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">@</td>
      <td style="text-align: left">@</td>
      <td style="text-align: left">Each sequence identifier line starts with @</td>
    </tr>
    <tr>
      <td style="text-align: left"><code class="highlighter-rouge">&lt;instrument&gt; </code></td>
      <td style="text-align: left">Characters allowed: a-z, A-Z, 0-9 and underscore</td>
      <td style="text-align: left">Instrument ID</td>
    </tr>
    <tr>
      <td style="text-align: left"><code class="highlighter-rouge">&lt;run number&gt;</code></td>
      <td style="text-align: left">Numerical</td>
      <td style="text-align: left">Run number on instrument</td>
    </tr>
    <tr>
      <td style="text-align: left"><code class="highlighter-rouge">&lt;flowcell ID&gt;</code></td>
      <td style="text-align: left">Characters allowed: a-z, A-Z, 0-9</td>
      <td style="text-align: left"> </td>
    </tr>
    <tr>
      <td style="text-align: left"><code class="highlighter-rouge">&lt;lane&gt;</code></td>
      <td style="text-align: left">Numerical</td>
      <td style="text-align: left">Lane number</td>
    </tr>
    <tr>
      <td style="text-align: left"><code class="highlighter-rouge">&lt;tile&gt;</code></td>
      <td style="text-align: left">Numerical</td>
      <td style="text-align: left">Tile number</td>
    </tr>
    <tr>
      <td style="text-align: left"><code class="highlighter-rouge">&lt;x_pos&gt;</code></td>
      <td style="text-align: left">Numerical</td>
      <td style="text-align: left">X coordinate of cluster</td>
    </tr>
    <tr>
      <td style="text-align: left"><code class="highlighter-rouge">&lt;y_pos&gt;</code></td>
      <td style="text-align: left">Numerical</td>
      <td style="text-align: left">Y coordinate of cluster</td>
    </tr>
    <tr>
      <td style="text-align: left"><code class="highlighter-rouge">&lt;read&gt;</code></td>
      <td style="text-align: left">Numerical</td>
      <td style="text-align: left">Read number. 1 can be single read or read 2 of paired-end</td>
    </tr>
    <tr>
      <td style="text-align: left"><code class="highlighter-rouge">&lt;is filtered&gt;</code></td>
      <td style="text-align: left">Y or N</td>
      <td style="text-align: left">Y if the read is filtered, N otherwise</td>
    </tr>
    <tr>
      <td style="text-align: left"><code class="highlighter-rouge">&lt;control number&gt;</code></td>
      <td style="text-align: left">Numerical</td>
      <td style="text-align: left">0 when none of the control bits are on, otherwise it is an even number.</td>
    </tr>
    <tr>
      <td style="text-align: left"><code class="highlighter-rouge">&lt;index sequence&gt;</code></td>
      <td style="text-align: left">ACTG</td>
      <td style="text-align: left">Index sequence</td>
    </tr>
  </tbody>
</table>

<p>因此<code class="highlighter-rouge">@EAS139:136:FC706VJ:2:5:1000:12850 1:Y:18:ATCACG</code>可以解释为：</p>

<ul>
  <li>测序仪id为<code class="highlighter-rouge">EAS139</code></li>
  <li>run number：136</li>
  <li>flowcell ID：FC706VJ</li>
  <li>lane：2</li>
  <li>tile：5</li>
  <li>x_pos：1000</li>
  <li>y_pos：12850</li>
  <li>read：1，代表是单端测序</li>
  <li>is filtered：Y，代表是filtered</li>
  <li>control number：18</li>
  <li>index sequence：ATCACG</li>
</ul>

<h2 id="phred-quality-score">Phred quality score</h2>

<blockquote>
  <p><a href="https://en.wikipedia.org/wiki/Phred_quality_score">A Phred quality score</a> is a measure of the quality of the identification of the nucleobases generated by automated DNA sequencing.[1][2] It was originally developed for Phred base calling to help in the automation of DNA sequencing in the Human Genome Project.</p>
</blockquote>

<p>Phred quality score是用来测定DNA自动测序中每个核算的测序质量的。它最早是为了人类基因组计划中的程序Phred base calling而开发的。</p>

<ul>
  <li>公式</li>
</ul>

<p><code class="highlighter-rouge">Q = -10 * log10(P)</code> &lt;==&gt; <code class="highlighter-rouge">P = 10 ^(-Q/10)</code></p>

<p>这里Q为Phred quality score，P为base-calling的error probabilities（错误率）。</p>

<ul>
  <li>具体表格</li>
</ul>

<table>
  <thead>
    <tr>
      <th style="text-align: left">Phred Quality Score</th>
      <th style="text-align: left">Probability of incorrect base call	（错误率）</th>
      <th style="text-align: left">Base call accuracy（准确率）</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">10</td>
      <td style="text-align: left">十分之一（1 in 10）</td>
      <td style="text-align: left">90%</td>
    </tr>
    <tr>
      <td style="text-align: left">20</td>
      <td style="text-align: left">百分之一（1 in 100）</td>
      <td style="text-align: left">99%</td>
    </tr>
    <tr>
      <td style="text-align: left">30</td>
      <td style="text-align: left">千分之一（1 in 1000）</td>
      <td style="text-align: left">99.9%</td>
    </tr>
    <tr>
      <td style="text-align: left">40</td>
      <td style="text-align: left">万分之一（1 in 10,000）</td>
      <td style="text-align: left">99.99%</td>
    </tr>
    <tr>
      <td style="text-align: left">50</td>
      <td style="text-align: left">十万分之一（1 in 100,000	）</td>
      <td style="text-align: left">99.999%</td>
    </tr>
    <tr>
      <td style="text-align: left">60</td>
      <td style="text-align: left">百万分之一（1 in 1,000,000）</td>
      <td style="text-align: left">99.9999%</td>
    </tr>
  </tbody>
</table>

<h2 id="encoding-of-quality-score">Encoding of Quality score</h2>

<table>
  <thead>
    <tr>
      <th style="text-align: left">类型</th>
      <th style="text-align: left">Quality score范围</th>
      <th style="text-align: left">ASCII范围</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">Sanger format</td>
      <td style="text-align: left">0 ~ 93（rarely exceeds 60）</td>
      <td style="text-align: left">33 ~ 126（Phred+33）</td>
    </tr>
    <tr>
      <td style="text-align: left">Solexa/Illumina 1.0 format</td>
      <td style="text-align: left">-5 ~ 40</td>
      <td style="text-align: left">59 ~ 104（Solexa+64）</td>
    </tr>
    <tr>
      <td style="text-align: left">Illumina 1.3+</td>
      <td style="text-align: left">0 ~ 40</td>
      <td style="text-align: left">64 ~ 104（Phred+64）</td>
    </tr>
    <tr>
      <td style="text-align: left">Illumina 1.5+</td>
      <td style="text-align: left">3 ~ 40</td>
      <td style="text-align: left">67 ~ 104（Phred+64）</td>
    </tr>
    <tr>
      <td style="text-align: left">Illumina 1.8+</td>
      <td style="text-align: left">0 ~ 41</td>
      <td style="text-align: left">33 ~ 74（Phred+33）</td>
    </tr>
  </tbody>
</table>

<blockquote>
  <p>For raw reads, the range of scores will depend on the technology and the base caller used, but will typically be up to 41 for recent Illumina chemistry</p>
</blockquote>

<p>对于原始的序列，质量值的范围取决于所用的技术和base caller，但是对于目前的Illumina技术，最多能达到41。</p>

<p>另外，常用软件<a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a>和<a href="http://www.usadellab.org/cms/?page=trimmomatic">Trimmomatic</a>均可以自动判断encoding的。</p>

<p>因此，质量值<code class="highlighter-rouge">BBBBCCCC?&lt;A?BC?7@@???????DBBA@@@@A@@</code>：</p>

<ul>
  <li>若采用Phred+64编码，应该为<code class="highlighter-rouge">2,2,2,6,6,6,6,6,6,6,6,6,6,9,9,9,9,9,9,9,9,9,9,9,9,9,9,2,6,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,6,6,6,6,6,6,6,6,6,6,2,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,2,6,6,6,6,6,6,6</code></li>
  <li>若采用Phred+32编码，应该为<code class="highlighter-rouge">33,33,33,37,37,37,37,37,37,37,37,37,37,40,40,40,40,40,40,40,40,40,40,40,40,40,40,33,37,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,37,37,37,37,37,37,37,37,37,37,33,37,37,37,37,37,37,37,37,37,37,37,37,37,37,37,37,37,37,37,37,37,37,37,37,37,37,33,37,37,37,37,37,37,37</code></li>
</ul>

<p>故，很明显，使用的是“Phred+32”编码，且质量值均大于30，错误率均小于千分之一，准确率大于99.9%。</p>

<h2 id="ascii">ASCII</h2>

<p><a href="https://en.wikipedia.org/wiki/ASCII">ASCII（American Standand Code for Information Interchange）</a>，是一套计算机编码系统。</p>

<p><img src="/images/ASCII-Table.png" alt="" /></p>

<h2 id="参考资料">参考资料</h2>
<ol>
  <li><a href="http://support.illumina.com/help/SequencingAnalysisWorkflow/Content/Vault/Informatics/Sequencing_Analysis/CASAVA/swSEQ_mCA_FASTQFiles.htm">Illumina support</a></li>
  <li><a href="https://en.wikipedia.org/wiki/FASTQ_format">FASTQ format on wiki</a></li>
  <li><a href="https://en.wikipedia.org/wiki/Phred_quality_score">Phred quality score on wiki</a></li>
  <li><a href="http://broadinstitute.github.io/picard/explain-qualities.html">explain qualities on picard</a></li>
  <li><a href="http://support.illumina.com/sequencing/sequencing_software/casava">Illumina CASAVA software</a></li>
  <li><a href="https://simple.wikipedia.org/wiki/ASCII">ASCII table</a></li>
</ol>

  </div>

</article>

      </div>
    </div>

    <footer class="site-footer">

  <div class="wrapper">


    <div class="footer-col-wrapper">
      <div class="footer-col footer-col-1">
        <ul class="contact-list">
          <li>zjuwhw's blog</li>
          <li><a href="mailto:zju_whw@163.com">zju_whw@163.com</a></li>
        </ul>
      </div>

      <div class="footer-col footer-col-2">
        <ul class="social-media-list">
          
          <li>
            <a href="https://github.com/zjuwhw"><span class="icon icon--github"><svg viewBox="0 0 16 16"><path fill="#828282" d="M7.999,0.431c-4.285,0-7.76,3.474-7.76,7.761 c0,3.428,2.223,6.337,5.307,7.363c0.388,0.071,0.53-0.168,0.53-0.374c0-0.184-0.007-0.672-0.01-1.32 c-2.159,0.469-2.614-1.04-2.614-1.04c-0.353-0.896-0.862-1.135-0.862-1.135c-0.705-0.481,0.053-0.472,0.053-0.472 c0.779,0.055,1.189,0.8,1.189,0.8c0.692,1.186,1.816,0.843,2.258,0.645c0.071-0.502,0.271-0.843,0.493-1.037 C4.86,11.425,3.049,10.76,3.049,7.786c0-0.847,0.302-1.54,0.799-2.082C3.768,5.507,3.501,4.718,3.924,3.65 c0,0,0.652-0.209,2.134,0.796C6.677,4.273,7.34,4.187,8,4.184c0.659,0.003,1.323,0.089,1.943,0.261 c1.482-1.004,2.132-0.796,2.132-0.796c0.423,1.068,0.157,1.857,0.077,2.054c0.497,0.542,0.798,1.235,0.798,2.082 c0,2.981-1.814,3.637-3.543,3.829c0.279,0.24,0.527,0.713,0.527,1.437c0,1.037-0.01,1.874-0.01,2.129 c0,0.208,0.14,0.449,0.534,0.373c3.081-1.028,5.302-3.935,5.302-7.362C15.76,3.906,12.285,0.431,7.999,0.431z"/></svg>
</span><span class="username">zjuwhw</span></a>

          </li>
          

          
          <li>
            <a href="https://twitter.com/zjuwhw"><span class="icon icon--twitter"><svg viewBox="0 0 16 16"><path fill="#828282" d="M15.969,3.058c-0.586,0.26-1.217,0.436-1.878,0.515c0.675-0.405,1.194-1.045,1.438-1.809c-0.632,0.375-1.332,0.647-2.076,0.793c-0.596-0.636-1.446-1.033-2.387-1.033c-1.806,0-3.27,1.464-3.27,3.27 c0,0.256,0.029,0.506,0.085,0.745C5.163,5.404,2.753,4.102,1.14,2.124C0.859,2.607,0.698,3.168,0.698,3.767 c0,1.134,0.577,2.135,1.455,2.722C1.616,6.472,1.112,6.325,0.671,6.08c0,0.014,0,0.027,0,0.041c0,1.584,1.127,2.906,2.623,3.206 C3.02,9.402,2.731,9.442,2.433,9.442c-0.211,0-0.416-0.021-0.615-0.059c0.416,1.299,1.624,2.245,3.055,2.271 c-1.119,0.877-2.529,1.4-4.061,1.4c-0.264,0-0.524-0.015-0.78-0.046c1.447,0.928,3.166,1.469,5.013,1.469 c6.015,0,9.304-4.983,9.304-9.304c0-0.142-0.003-0.283-0.009-0.423C14.976,4.29,15.531,3.714,15.969,3.058z"/></svg>
</span><span class="username">zjuwhw</span></a>

          </li>
          
        </ul>
      </div>

      <div class="footer-col footer-col-3">
        <p>This is zjuwhw's personal blog. Powered by Jekyll and Host in Github.
</p>
      </div>
    </div>

  </div>

</footer>


  </body>

</html>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    processEscapes: true
  }
});
</script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js?config=TeX-AMS-MML_HTMLorMML" type="text/javascript"></script>
